// @aws-sdk/client-kms/dist-cjs/KMS.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KMS = void 0;
const CancelKeyDeletionCommand_1 = require("./commands/CancelKeyDeletionCommand");
const ConnectCustomKeyStoreCommand_1 = require("./commands/ConnectCustomKeyStoreCommand");
const CreateAliasCommand_1 = require("./commands/CreateAliasCommand");
const CreateCustomKeyStoreCommand_1 = require("./commands/CreateCustomKeyStoreCommand");
const CreateGrantCommand_1 = require("./commands/CreateGrantCommand");
const CreateKeyCommand_1 = require("./commands/CreateKeyCommand");
const DecryptCommand_1 = require("./commands/DecryptCommand");
const DeleteAliasCommand_1 = require("./commands/DeleteAliasCommand");
const DeleteCustomKeyStoreCommand_1 = require("./commands/DeleteCustomKeyStoreCommand");
const DeleteImportedKeyMaterialCommand_1 = require("./commands/DeleteImportedKeyMaterialCommand");
const DescribeCustomKeyStoresCommand_1 = require("./commands/DescribeCustomKeyStoresCommand");
const DescribeKeyCommand_1 = require("./commands/DescribeKeyCommand");
const DisableKeyCommand_1 = require("./commands/DisableKeyCommand");
const DisableKeyRotationCommand_1 = require("./commands/DisableKeyRotationCommand");
const DisconnectCustomKeyStoreCommand_1 = require("./commands/DisconnectCustomKeyStoreCommand");
const EnableKeyCommand_1 = require("./commands/EnableKeyCommand");
const EnableKeyRotationCommand_1 = require("./commands/EnableKeyRotationCommand");
const EncryptCommand_1 = require("./commands/EncryptCommand");
const GenerateDataKeyCommand_1 = require("./commands/GenerateDataKeyCommand");
const GenerateDataKeyPairCommand_1 = require("./commands/GenerateDataKeyPairCommand");
const GenerateDataKeyPairWithoutPlaintextCommand_1 = require("./commands/GenerateDataKeyPairWithoutPlaintextCommand");
const GenerateDataKeyWithoutPlaintextCommand_1 = require("./commands/GenerateDataKeyWithoutPlaintextCommand");
const GenerateRandomCommand_1 = require("./commands/GenerateRandomCommand");
const GetKeyPolicyCommand_1 = require("./commands/GetKeyPolicyCommand");
const GetKeyRotationStatusCommand_1 = require("./commands/GetKeyRotationStatusCommand");
const GetParametersForImportCommand_1 = require("./commands/GetParametersForImportCommand");
const GetPublicKeyCommand_1 = require("./commands/GetPublicKeyCommand");
const ImportKeyMaterialCommand_1 = require("./commands/ImportKeyMaterialCommand");
const ListAliasesCommand_1 = require("./commands/ListAliasesCommand");
const ListGrantsCommand_1 = require("./commands/ListGrantsCommand");
const ListKeyPoliciesCommand_1 = require("./commands/ListKeyPoliciesCommand");
const ListKeysCommand_1 = require("./commands/ListKeysCommand");
const ListResourceTagsCommand_1 = require("./commands/ListResourceTagsCommand");
const ListRetirableGrantsCommand_1 = require("./commands/ListRetirableGrantsCommand");
const PutKeyPolicyCommand_1 = require("./commands/PutKeyPolicyCommand");
const ReEncryptCommand_1 = require("./commands/ReEncryptCommand");
const ReplicateKeyCommand_1 = require("./commands/ReplicateKeyCommand");
const RetireGrantCommand_1 = require("./commands/RetireGrantCommand");
const RevokeGrantCommand_1 = require("./commands/RevokeGrantCommand");
const ScheduleKeyDeletionCommand_1 = require("./commands/ScheduleKeyDeletionCommand");
const SignCommand_1 = require("./commands/SignCommand");
const TagResourceCommand_1 = require("./commands/TagResourceCommand");
const UntagResourceCommand_1 = require("./commands/UntagResourceCommand");
const UpdateAliasCommand_1 = require("./commands/UpdateAliasCommand");
const UpdateCustomKeyStoreCommand_1 = require("./commands/UpdateCustomKeyStoreCommand");
const UpdateKeyDescriptionCommand_1 = require("./commands/UpdateKeyDescriptionCommand");
const UpdatePrimaryRegionCommand_1 = require("./commands/UpdatePrimaryRegionCommand");
const VerifyCommand_1 = require("./commands/VerifyCommand");
const KMSClient_1 = require("./KMSClient");
class KMS extends KMSClient_1.KMSClient {
    cancelKeyDeletion(args, optionsOrCb, cb) {
        const command = new CancelKeyDeletionCommand_1.CancelKeyDeletionCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    connectCustomKeyStore(args, optionsOrCb, cb) {
        const command = new ConnectCustomKeyStoreCommand_1.ConnectCustomKeyStoreCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    createAlias(args, optionsOrCb, cb) {
        const command = new CreateAliasCommand_1.CreateAliasCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    createCustomKeyStore(args, optionsOrCb, cb) {
        const command = new CreateCustomKeyStoreCommand_1.CreateCustomKeyStoreCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    createGrant(args, optionsOrCb, cb) {
        const command = new CreateGrantCommand_1.CreateGrantCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    createKey(args, optionsOrCb, cb) {
        const command = new CreateKeyCommand_1.CreateKeyCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    decrypt(args, optionsOrCb, cb) {
        const command = new DecryptCommand_1.DecryptCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    deleteAlias(args, optionsOrCb, cb) {
        const command = new DeleteAliasCommand_1.DeleteAliasCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    deleteCustomKeyStore(args, optionsOrCb, cb) {
        const command = new DeleteCustomKeyStoreCommand_1.DeleteCustomKeyStoreCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    deleteImportedKeyMaterial(args, optionsOrCb, cb) {
        const command = new DeleteImportedKeyMaterialCommand_1.DeleteImportedKeyMaterialCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    describeCustomKeyStores(args, optionsOrCb, cb) {
        const command = new DescribeCustomKeyStoresCommand_1.DescribeCustomKeyStoresCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    describeKey(args, optionsOrCb, cb) {
        const command = new DescribeKeyCommand_1.DescribeKeyCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    disableKey(args, optionsOrCb, cb) {
        const command = new DisableKeyCommand_1.DisableKeyCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    disableKeyRotation(args, optionsOrCb, cb) {
        const command = new DisableKeyRotationCommand_1.DisableKeyRotationCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    disconnectCustomKeyStore(args, optionsOrCb, cb) {
        const command = new DisconnectCustomKeyStoreCommand_1.DisconnectCustomKeyStoreCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    enableKey(args, optionsOrCb, cb) {
        const command = new EnableKeyCommand_1.EnableKeyCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    enableKeyRotation(args, optionsOrCb, cb) {
        const command = new EnableKeyRotationCommand_1.EnableKeyRotationCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    encrypt(args, optionsOrCb, cb) {
        const command = new EncryptCommand_1.EncryptCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    generateDataKey(args, optionsOrCb, cb) {
        const command = new GenerateDataKeyCommand_1.GenerateDataKeyCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    generateDataKeyPair(args, optionsOrCb, cb) {
        const command = new GenerateDataKeyPairCommand_1.GenerateDataKeyPairCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    generateDataKeyPairWithoutPlaintext(args, optionsOrCb, cb) {
        const command = new GenerateDataKeyPairWithoutPlaintextCommand_1.GenerateDataKeyPairWithoutPlaintextCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    generateDataKeyWithoutPlaintext(args, optionsOrCb, cb) {
        const command = new GenerateDataKeyWithoutPlaintextCommand_1.GenerateDataKeyWithoutPlaintextCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    generateRandom(args, optionsOrCb, cb) {
        const command = new GenerateRandomCommand_1.GenerateRandomCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    getKeyPolicy(args, optionsOrCb, cb) {
        const command = new GetKeyPolicyCommand_1.GetKeyPolicyCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    getKeyRotationStatus(args, optionsOrCb, cb) {
        const command = new GetKeyRotationStatusCommand_1.GetKeyRotationStatusCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    getParametersForImport(args, optionsOrCb, cb) {
        const command = new GetParametersForImportCommand_1.GetParametersForImportCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    getPublicKey(args, optionsOrCb, cb) {
        const command = new GetPublicKeyCommand_1.GetPublicKeyCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    importKeyMaterial(args, optionsOrCb, cb) {
        const command = new ImportKeyMaterialCommand_1.ImportKeyMaterialCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    listAliases(args, optionsOrCb, cb) {
        const command = new ListAliasesCommand_1.ListAliasesCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    listGrants(args, optionsOrCb, cb) {
        const command = new ListGrantsCommand_1.ListGrantsCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    listKeyPolicies(args, optionsOrCb, cb) {
        const command = new ListKeyPoliciesCommand_1.ListKeyPoliciesCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    listKeys(args, optionsOrCb, cb) {
        const command = new ListKeysCommand_1.ListKeysCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    listResourceTags(args, optionsOrCb, cb) {
        const command = new ListResourceTagsCommand_1.ListResourceTagsCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    listRetirableGrants(args, optionsOrCb, cb) {
        const command = new ListRetirableGrantsCommand_1.ListRetirableGrantsCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    putKeyPolicy(args, optionsOrCb, cb) {
        const command = new PutKeyPolicyCommand_1.PutKeyPolicyCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    reEncrypt(args, optionsOrCb, cb) {
        const command = new ReEncryptCommand_1.ReEncryptCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    replicateKey(args, optionsOrCb, cb) {
        const command = new ReplicateKeyCommand_1.ReplicateKeyCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    retireGrant(args, optionsOrCb, cb) {
        const command = new RetireGrantCommand_1.RetireGrantCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    revokeGrant(args, optionsOrCb, cb) {
        const command = new RevokeGrantCommand_1.RevokeGrantCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    scheduleKeyDeletion(args, optionsOrCb, cb) {
        const command = new ScheduleKeyDeletionCommand_1.ScheduleKeyDeletionCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    sign(args, optionsOrCb, cb) {
        const command = new SignCommand_1.SignCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    tagResource(args, optionsOrCb, cb) {
        const command = new TagResourceCommand_1.TagResourceCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    untagResource(args, optionsOrCb, cb) {
        const command = new UntagResourceCommand_1.UntagResourceCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    updateAlias(args, optionsOrCb, cb) {
        const command = new UpdateAliasCommand_1.UpdateAliasCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    updateCustomKeyStore(args, optionsOrCb, cb) {
        const command = new UpdateCustomKeyStoreCommand_1.UpdateCustomKeyStoreCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    updateKeyDescription(args, optionsOrCb, cb) {
        const command = new UpdateKeyDescriptionCommand_1.UpdateKeyDescriptionCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    updatePrimaryRegion(args, optionsOrCb, cb) {
        const command = new UpdatePrimaryRegionCommand_1.UpdatePrimaryRegionCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    verify(args, optionsOrCb, cb) {
        const command = new VerifyCommand_1.VerifyCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
}
exports.KMS = KMS;
