// @aws-sdk/client-kms/dist-es/commands/ConnectCustomKeyStoreCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ConnectCustomKeyStoreRequest, ConnectCustomKeyStoreResponse } from "../models/models_0";
import { deserializeAws_json1_1ConnectCustomKeyStoreCommand, serializeAws_json1_1ConnectCustomKeyStoreCommand, } from "../protocols/Aws_json1_1";
var ConnectCustomKeyStoreCommand = (function (_super) {
    __extends(ConnectCustomKeyStoreCommand, _super);
    function ConnectCustomKeyStoreCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ConnectCustomKeyStoreCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "KMSClient";
        var commandName = "ConnectCustomKeyStoreCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ConnectCustomKeyStoreRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ConnectCustomKeyStoreResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ConnectCustomKeyStoreCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ConnectCustomKeyStoreCommand(input, context);
    };
    ConnectCustomKeyStoreCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ConnectCustomKeyStoreCommand(output, context);
    };
    return ConnectCustomKeyStoreCommand;
}($Command));
export { ConnectCustomKeyStoreCommand };
