// @aws-sdk/client-kms/dist-es/commands/CreateCustomKeyStoreCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateCustomKeyStoreRequest, CreateCustomKeyStoreResponse } from "../models/models_0";
import { deserializeAws_json1_1CreateCustomKeyStoreCommand, serializeAws_json1_1CreateCustomKeyStoreCommand, } from "../protocols/Aws_json1_1";
var CreateCustomKeyStoreCommand = (function (_super) {
    __extends(CreateCustomKeyStoreCommand, _super);
    function CreateCustomKeyStoreCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateCustomKeyStoreCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "KMSClient";
        var commandName = "CreateCustomKeyStoreCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateCustomKeyStoreRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateCustomKeyStoreResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateCustomKeyStoreCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1CreateCustomKeyStoreCommand(input, context);
    };
    CreateCustomKeyStoreCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1CreateCustomKeyStoreCommand(output, context);
    };
    return CreateCustomKeyStoreCommand;
}($Command));
export { CreateCustomKeyStoreCommand };
