// @aws-sdk/client-kms/dist-es/commands/DecryptCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DecryptRequest, DecryptResponse } from "../models/models_0";
import { deserializeAws_json1_1DecryptCommand, serializeAws_json1_1DecryptCommand } from "../protocols/Aws_json1_1";
var DecryptCommand = (function (_super) {
    __extends(DecryptCommand, _super);
    function DecryptCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DecryptCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "KMSClient";
        var commandName = "DecryptCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DecryptRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DecryptResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DecryptCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DecryptCommand(input, context);
    };
    DecryptCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DecryptCommand(output, context);
    };
    return DecryptCommand;
}($Command));
export { DecryptCommand };
