// @aws-sdk/client-kms/dist-es/commands/DescribeCustomKeyStoresCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeCustomKeyStoresRequest, DescribeCustomKeyStoresResponse } from "../models/models_0";
import { deserializeAws_json1_1DescribeCustomKeyStoresCommand, serializeAws_json1_1DescribeCustomKeyStoresCommand, } from "../protocols/Aws_json1_1";
var DescribeCustomKeyStoresCommand = (function (_super) {
    __extends(DescribeCustomKeyStoresCommand, _super);
    function DescribeCustomKeyStoresCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeCustomKeyStoresCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "KMSClient";
        var commandName = "DescribeCustomKeyStoresCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeCustomKeyStoresRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeCustomKeyStoresResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeCustomKeyStoresCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DescribeCustomKeyStoresCommand(input, context);
    };
    DescribeCustomKeyStoresCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DescribeCustomKeyStoresCommand(output, context);
    };
    return DescribeCustomKeyStoresCommand;
}($Command));
export { DescribeCustomKeyStoresCommand };
