// @aws-sdk/client-kms/dist-es/commands/DescribeKeyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeKeyRequest, DescribeKeyResponse } from "../models/models_0";
import { deserializeAws_json1_1DescribeKeyCommand, serializeAws_json1_1DescribeKeyCommand, } from "../protocols/Aws_json1_1";
var DescribeKeyCommand = (function (_super) {
    __extends(DescribeKeyCommand, _super);
    function DescribeKeyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeKeyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "KMSClient";
        var commandName = "DescribeKeyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeKeyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeKeyResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeKeyCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DescribeKeyCommand(input, context);
    };
    DescribeKeyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DescribeKeyCommand(output, context);
    };
    return DescribeKeyCommand;
}($Command));
export { DescribeKeyCommand };
