// @aws-sdk/client-kms/dist-es/commands/DisableKeyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DisableKeyRequest } from "../models/models_0";
import { deserializeAws_json1_1DisableKeyCommand, serializeAws_json1_1DisableKeyCommand, } from "../protocols/Aws_json1_1";
var DisableKeyCommand = (function (_super) {
    __extends(DisableKeyCommand, _super);
    function DisableKeyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DisableKeyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "KMSClient";
        var commandName = "DisableKeyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DisableKeyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DisableKeyCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DisableKeyCommand(input, context);
    };
    DisableKeyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DisableKeyCommand(output, context);
    };
    return DisableKeyCommand;
}($Command));
export { DisableKeyCommand };
