// @aws-sdk/client-kms/dist-es/commands/GenerateDataKeyWithoutPlaintextCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GenerateDataKeyWithoutPlaintextRequest, GenerateDataKeyWithoutPlaintextResponse } from "../models/models_0";
import { deserializeAws_json1_1GenerateDataKeyWithoutPlaintextCommand, serializeAws_json1_1GenerateDataKeyWithoutPlaintextCommand, } from "../protocols/Aws_json1_1";
var GenerateDataKeyWithoutPlaintextCommand = (function (_super) {
    __extends(GenerateDataKeyWithoutPlaintextCommand, _super);
    function GenerateDataKeyWithoutPlaintextCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GenerateDataKeyWithoutPlaintextCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "KMSClient";
        var commandName = "GenerateDataKeyWithoutPlaintextCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GenerateDataKeyWithoutPlaintextRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GenerateDataKeyWithoutPlaintextResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GenerateDataKeyWithoutPlaintextCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GenerateDataKeyWithoutPlaintextCommand(input, context);
    };
    GenerateDataKeyWithoutPlaintextCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GenerateDataKeyWithoutPlaintextCommand(output, context);
    };
    return GenerateDataKeyWithoutPlaintextCommand;
}($Command));
export { GenerateDataKeyWithoutPlaintextCommand };
