// @aws-sdk/client-kms/dist-es/commands/GetKeyPolicyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetKeyPolicyRequest, GetKeyPolicyResponse } from "../models/models_0";
import { deserializeAws_json1_1GetKeyPolicyCommand, serializeAws_json1_1GetKeyPolicyCommand, } from "../protocols/Aws_json1_1";
var GetKeyPolicyCommand = (function (_super) {
    __extends(GetKeyPolicyCommand, _super);
    function GetKeyPolicyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetKeyPolicyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "KMSClient";
        var commandName = "GetKeyPolicyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetKeyPolicyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetKeyPolicyResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetKeyPolicyCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GetKeyPolicyCommand(input, context);
    };
    GetKeyPolicyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GetKeyPolicyCommand(output, context);
    };
    return GetKeyPolicyCommand;
}($Command));
export { GetKeyPolicyCommand };
