// @aws-sdk/client-kms/dist-es/commands/GetParametersForImportCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetParametersForImportRequest, GetParametersForImportResponse } from "../models/models_0";
import { deserializeAws_json1_1GetParametersForImportCommand, serializeAws_json1_1GetParametersForImportCommand, } from "../protocols/Aws_json1_1";
var GetParametersForImportCommand = (function (_super) {
    __extends(GetParametersForImportCommand, _super);
    function GetParametersForImportCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetParametersForImportCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "KMSClient";
        var commandName = "GetParametersForImportCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetParametersForImportRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetParametersForImportResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetParametersForImportCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GetParametersForImportCommand(input, context);
    };
    GetParametersForImportCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GetParametersForImportCommand(output, context);
    };
    return GetParametersForImportCommand;
}($Command));
export { GetParametersForImportCommand };
