// @aws-sdk/client-kms/dist-es/commands/GetPublicKeyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetPublicKeyRequest, GetPublicKeyResponse } from "../models/models_0";
import { deserializeAws_json1_1GetPublicKeyCommand, serializeAws_json1_1GetPublicKeyCommand, } from "../protocols/Aws_json1_1";
var GetPublicKeyCommand = (function (_super) {
    __extends(GetPublicKeyCommand, _super);
    function GetPublicKeyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetPublicKeyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "KMSClient";
        var commandName = "GetPublicKeyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetPublicKeyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetPublicKeyResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetPublicKeyCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GetPublicKeyCommand(input, context);
    };
    GetPublicKeyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GetPublicKeyCommand(output, context);
    };
    return GetPublicKeyCommand;
}($Command));
export { GetPublicKeyCommand };
