// @aws-sdk/client-kms/dist-es/commands/ListAliasesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListAliasesRequest, ListAliasesResponse } from "../models/models_0";
import { deserializeAws_json1_1ListAliasesCommand, serializeAws_json1_1ListAliasesCommand, } from "../protocols/Aws_json1_1";
var ListAliasesCommand = (function (_super) {
    __extends(ListAliasesCommand, _super);
    function ListAliasesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListAliasesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "KMSClient";
        var commandName = "ListAliasesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListAliasesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListAliasesResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListAliasesCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListAliasesCommand(input, context);
    };
    ListAliasesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListAliasesCommand(output, context);
    };
    return ListAliasesCommand;
}($Command));
export { ListAliasesCommand };
