// @aws-sdk/client-kms/dist-es/commands/ListGrantsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListGrantsRequest, ListGrantsResponse } from "../models/models_0";
import { deserializeAws_json1_1ListGrantsCommand, serializeAws_json1_1ListGrantsCommand, } from "../protocols/Aws_json1_1";
var ListGrantsCommand = (function (_super) {
    __extends(ListGrantsCommand, _super);
    function ListGrantsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListGrantsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "KMSClient";
        var commandName = "ListGrantsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListGrantsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListGrantsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListGrantsCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListGrantsCommand(input, context);
    };
    ListGrantsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListGrantsCommand(output, context);
    };
    return ListGrantsCommand;
}($Command));
export { ListGrantsCommand };
