// @aws-sdk/client-kms/dist-es/commands/ReEncryptCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ReEncryptRequest, ReEncryptResponse } from "../models/models_0";
import { deserializeAws_json1_1ReEncryptCommand, serializeAws_json1_1ReEncryptCommand } from "../protocols/Aws_json1_1";
var ReEncryptCommand = (function (_super) {
    __extends(ReEncryptCommand, _super);
    function ReEncryptCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ReEncryptCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "KMSClient";
        var commandName = "ReEncryptCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ReEncryptRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ReEncryptResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ReEncryptCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ReEncryptCommand(input, context);
    };
    ReEncryptCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ReEncryptCommand(output, context);
    };
    return ReEncryptCommand;
}($Command));
export { ReEncryptCommand };
