// @aws-sdk/client-kms/dist-es/commands/ReplicateKeyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ReplicateKeyRequest, ReplicateKeyResponse } from "../models/models_0";
import { deserializeAws_json1_1ReplicateKeyCommand, serializeAws_json1_1ReplicateKeyCommand, } from "../protocols/Aws_json1_1";
var ReplicateKeyCommand = (function (_super) {
    __extends(ReplicateKeyCommand, _super);
    function ReplicateKeyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ReplicateKeyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "KMSClient";
        var commandName = "ReplicateKeyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ReplicateKeyRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ReplicateKeyResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ReplicateKeyCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ReplicateKeyCommand(input, context);
    };
    ReplicateKeyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ReplicateKeyCommand(output, context);
    };
    return ReplicateKeyCommand;
}($Command));
export { ReplicateKeyCommand };
