// @aws-sdk/client-kms/dist-es/commands/SignCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { SignRequest, SignResponse } from "../models/models_0";
import { deserializeAws_json1_1SignCommand, serializeAws_json1_1SignCommand } from "../protocols/Aws_json1_1";
var SignCommand = (function (_super) {
    __extends(SignCommand, _super);
    function SignCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    SignCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "KMSClient";
        var commandName = "SignCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: SignRequest.filterSensitiveLog,
            outputFilterSensitiveLog: SignResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    SignCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1SignCommand(input, context);
    };
    SignCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1SignCommand(output, context);
    };
    return SignCommand;
}($Command));
export { SignCommand };
