// @aws-sdk/client-kms/dist-es/commands/UpdatePrimaryRegionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdatePrimaryRegionRequest } from "../models/models_0";
import { deserializeAws_json1_1UpdatePrimaryRegionCommand, serializeAws_json1_1UpdatePrimaryRegionCommand, } from "../protocols/Aws_json1_1";
var UpdatePrimaryRegionCommand = (function (_super) {
    __extends(UpdatePrimaryRegionCommand, _super);
    function UpdatePrimaryRegionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdatePrimaryRegionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "KMSClient";
        var commandName = "UpdatePrimaryRegionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdatePrimaryRegionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdatePrimaryRegionCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1UpdatePrimaryRegionCommand(input, context);
    };
    UpdatePrimaryRegionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1UpdatePrimaryRegionCommand(output, context);
    };
    return UpdatePrimaryRegionCommand;
}($Command));
export { UpdatePrimaryRegionCommand };
