// @aws-sdk/client-kms/dist-cjs/models/models_0.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateKeyResponse = exports.KeyMetadata = exports.SigningAlgorithmSpec = exports.MultiRegionConfiguration = exports.MultiRegionKey = exports.MultiRegionKeyType = exports.KeyState = exports.KeyManagerType = exports.ExpirationModelType = exports.EncryptionAlgorithmSpec = exports.CreateKeyRequest = exports.Tag = exports.OriginType = exports.KeyUsageType = exports.KeySpec = exports.CustomerMasterKeySpec = exports.InvalidGrantTokenException = exports.DisabledException = exports.CreateGrantResponse = exports.CreateGrantRequest = exports.GrantOperation = exports.GrantConstraints = exports.IncorrectTrustAnchorException = exports.CustomKeyStoreNameInUseException = exports.CreateCustomKeyStoreResponse = exports.CreateCustomKeyStoreRequest = exports.LimitExceededException = exports.InvalidAliasNameException = exports.CreateAliasRequest = exports.ConnectionStateType = exports.ConnectionErrorCodeType = exports.CustomKeyStoreNotFoundException = exports.CustomKeyStoreInvalidStateException = exports.ConnectCustomKeyStoreResponse = exports.ConnectCustomKeyStoreRequest = exports.CloudHsmClusterNotRelatedException = exports.CloudHsmClusterNotFoundException = exports.CloudHsmClusterNotActiveException = exports.CloudHsmClusterInvalidConfigurationException = exports.CloudHsmClusterInUseException = exports.NotFoundException = exports.KMSInvalidStateException = exports.KMSInternalException = exports.InvalidArnException = exports.DependencyTimeoutException = exports.CancelKeyDeletionResponse = exports.CancelKeyDeletionRequest = exports.AlreadyExistsException = exports.AliasListEntry = exports.AlgorithmSpec = void 0;
exports.GetPublicKeyResponse = exports.GetPublicKeyRequest = exports.GetParametersForImportResponse = exports.GetParametersForImportRequest = exports.WrappingKeySpec = exports.GetKeyRotationStatusResponse = exports.GetKeyRotationStatusRequest = exports.GetKeyPolicyResponse = exports.GetKeyPolicyRequest = exports.GenerateRandomResponse = exports.GenerateRandomRequest = exports.GenerateDataKeyWithoutPlaintextResponse = exports.GenerateDataKeyWithoutPlaintextRequest = exports.GenerateDataKeyPairWithoutPlaintextResponse = exports.GenerateDataKeyPairWithoutPlaintextRequest = exports.GenerateDataKeyPairResponse = exports.GenerateDataKeyPairRequest = exports.GenerateDataKeyResponse = exports.GenerateDataKeyRequest = exports.ExpiredImportTokenException = exports.EncryptResponse = exports.EncryptRequest = exports.EnableKeyRotationRequest = exports.EnableKeyRequest = exports.DisconnectCustomKeyStoreResponse = exports.DisconnectCustomKeyStoreRequest = exports.DisableKeyRotationRequest = exports.DisableKeyRequest = exports.DescribeKeyResponse = exports.DescribeKeyRequest = exports.InvalidMarkerException = exports.DescribeCustomKeyStoresResponse = exports.DescribeCustomKeyStoresRequest = exports.DeleteImportedKeyMaterialRequest = exports.DeleteCustomKeyStoreResponse = exports.DeleteCustomKeyStoreRequest = exports.DeleteAliasRequest = exports.KeyUnavailableException = exports.InvalidKeyUsageException = exports.InvalidCiphertextException = exports.IncorrectKeyException = exports.DecryptResponse = exports.DecryptRequest = exports.DataKeySpec = exports.DataKeyPairSpec = exports.CustomKeyStoresListEntry = exports.CustomKeyStoreHasCMKsException = exports.UnsupportedOperationException = exports.TagException = exports.MalformedPolicyDocumentException = void 0;
exports.VerifyResponse = exports.VerifyRequest = exports.UpdatePrimaryRegionRequest = exports.UpdateKeyDescriptionRequest = exports.UpdateCustomKeyStoreResponse = exports.UpdateCustomKeyStoreRequest = exports.UpdateAliasRequest = exports.UntagResourceRequest = exports.TagResourceRequest = exports.SignResponse = exports.SignRequest = exports.ScheduleKeyDeletionResponse = exports.ScheduleKeyDeletionRequest = exports.RevokeGrantRequest = exports.RetireGrantRequest = exports.ReplicateKeyResponse = exports.ReplicateKeyRequest = exports.ReEncryptResponse = exports.ReEncryptRequest = exports.PutKeyPolicyRequest = exports.MessageType = exports.ListRetirableGrantsRequest = exports.ListResourceTagsResponse = exports.ListResourceTagsRequest = exports.ListKeysResponse = exports.ListKeysRequest = exports.ListKeyPoliciesResponse = exports.ListKeyPoliciesRequest = exports.ListGrantsResponse = exports.ListGrantsRequest = exports.ListAliasesResponse = exports.ListAliasesRequest = exports.KMSInvalidSignatureException = exports.KeyListEntry = exports.InvalidGrantIdException = exports.InvalidImportTokenException = exports.IncorrectKeyMaterialException = exports.ImportKeyMaterialResponse = exports.ImportKeyMaterialRequest = exports.GrantListEntry = void 0;
const smithy_client_1 = require("@aws-sdk/smithy-client");
const KMSServiceException_1 = require("./KMSServiceException");
var AlgorithmSpec;
(function (AlgorithmSpec) {
    AlgorithmSpec["RSAES_OAEP_SHA_1"] = "RSAES_OAEP_SHA_1";
    AlgorithmSpec["RSAES_OAEP_SHA_256"] = "RSAES_OAEP_SHA_256";
    AlgorithmSpec["RSAES_PKCS1_V1_5"] = "RSAES_PKCS1_V1_5";
})(AlgorithmSpec = exports.AlgorithmSpec || (exports.AlgorithmSpec = {}));
var AliasListEntry;
(function (AliasListEntry) {
    AliasListEntry.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AliasListEntry = exports.AliasListEntry || (exports.AliasListEntry = {}));
class AlreadyExistsException extends KMSServiceException_1.KMSServiceException {
    constructor(opts) {
        super({
            name: "AlreadyExistsException",
            $fault: "client",
            ...opts,
        });
        this.name = "AlreadyExistsException";
        this.$fault = "client";
        Object.setPrototypeOf(this, AlreadyExistsException.prototype);
    }
}
exports.AlreadyExistsException = AlreadyExistsException;
var CancelKeyDeletionRequest;
(function (CancelKeyDeletionRequest) {
    CancelKeyDeletionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CancelKeyDeletionRequest = exports.CancelKeyDeletionRequest || (exports.CancelKeyDeletionRequest = {}));
var CancelKeyDeletionResponse;
(function (CancelKeyDeletionResponse) {
    CancelKeyDeletionResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CancelKeyDeletionResponse = exports.CancelKeyDeletionResponse || (exports.CancelKeyDeletionResponse = {}));
class DependencyTimeoutException extends KMSServiceException_1.KMSServiceException {
    constructor(opts) {
        super({
            name: "DependencyTimeoutException",
            $fault: "server",
            ...opts,
        });
        this.name = "DependencyTimeoutException";
        this.$fault = "server";
        Object.setPrototypeOf(this, DependencyTimeoutException.prototype);
    }
}
exports.DependencyTimeoutException = DependencyTimeoutException;
class InvalidArnException extends KMSServiceException_1.KMSServiceException {
    constructor(opts) {
        super({
            name: "InvalidArnException",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidArnException";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidArnException.prototype);
    }
}
exports.InvalidArnException = InvalidArnException;
class KMSInternalException extends KMSServiceException_1.KMSServiceException {
    constructor(opts) {
        super({
            name: "KMSInternalException",
            $fault: "server",
            ...opts,
        });
        this.name = "KMSInternalException";
        this.$fault = "server";
        Object.setPrototypeOf(this, KMSInternalException.prototype);
    }
}
exports.KMSInternalException = KMSInternalException;
class KMSInvalidStateException extends KMSServiceException_1.KMSServiceException {
    constructor(opts) {
        super({
            name: "KMSInvalidStateException",
            $fault: "client",
            ...opts,
        });
        this.name = "KMSInvalidStateException";
        this.$fault = "client";
        Object.setPrototypeOf(this, KMSInvalidStateException.prototype);
    }
}
exports.KMSInvalidStateException = KMSInvalidStateException;
class NotFoundException extends KMSServiceException_1.KMSServiceException {
    constructor(opts) {
        super({
            name: "NotFoundException",
            $fault: "client",
            ...opts,
        });
        this.name = "NotFoundException";
        this.$fault = "client";
        Object.setPrototypeOf(this, NotFoundException.prototype);
    }
}
exports.NotFoundException = NotFoundException;
class CloudHsmClusterInUseException extends KMSServiceException_1.KMSServiceException {
    constructor(opts) {
        super({
            name: "CloudHsmClusterInUseException",
            $fault: "client",
            ...opts,
        });
        this.name = "CloudHsmClusterInUseException";
        this.$fault = "client";
        Object.setPrototypeOf(this, CloudHsmClusterInUseException.prototype);
    }
}
exports.CloudHsmClusterInUseException = CloudHsmClusterInUseException;
class CloudHsmClusterInvalidConfigurationException extends KMSServiceException_1.KMSServiceException {
    constructor(opts) {
        super({
            name: "CloudHsmClusterInvalidConfigurationException",
            $fault: "client",
            ...opts,
        });
        this.name = "CloudHsmClusterInvalidConfigurationException";
        this.$fault = "client";
        Object.setPrototypeOf(this, CloudHsmClusterInvalidConfigurationException.prototype);
    }
}
exports.CloudHsmClusterInvalidConfigurationException = CloudHsmClusterInvalidConfigurationException;
class CloudHsmClusterNotActiveException extends KMSServiceException_1.KMSServiceException {
    constructor(opts) {
        super({
            name: "CloudHsmClusterNotActiveException",
            $fault: "client",
            ...opts,
        });
        this.name = "CloudHsmClusterNotActiveException";
        this.$fault = "client";
        Object.setPrototypeOf(this, CloudHsmClusterNotActiveException.prototype);
    }
}
exports.CloudHsmClusterNotActiveException = CloudHsmClusterNotActiveException;
class CloudHsmClusterNotFoundException extends KMSServiceException_1.KMSServiceException {
    constructor(opts) {
        super({
            name: "CloudHsmClusterNotFoundException",
            $fault: "client",
            ...opts,
        });
        this.name = "CloudHsmClusterNotFoundException";
        this.$fault = "client";
        Object.setPrototypeOf(this, CloudHsmClusterNotFoundException.prototype);
    }
}
exports.CloudHsmClusterNotFoundException = CloudHsmClusterNotFoundException;
class CloudHsmClusterNotRelatedException extends KMSServiceException_1.KMSServiceException {
    constructor(opts) {
        super({
            name: "CloudHsmClusterNotRelatedException",
            $fault: "client",
            ...opts,
        });
        this.name = "CloudHsmClusterNotRelatedException";
        this.$fault = "client";
        Object.setPrototypeOf(this, CloudHsmClusterNotRelatedException.prototype);
    }
}
exports.CloudHsmClusterNotRelatedException = CloudHsmClusterNotRelatedException;
var ConnectCustomKeyStoreRequest;
(function (ConnectCustomKeyStoreRequest) {
    ConnectCustomKeyStoreRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ConnectCustomKeyStoreRequest = exports.ConnectCustomKeyStoreRequest || (exports.ConnectCustomKeyStoreRequest = {}));
var ConnectCustomKeyStoreResponse;
(function (ConnectCustomKeyStoreResponse) {
    ConnectCustomKeyStoreResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ConnectCustomKeyStoreResponse = exports.ConnectCustomKeyStoreResponse || (exports.ConnectCustomKeyStoreResponse = {}));
class CustomKeyStoreInvalidStateException extends KMSServiceException_1.KMSServiceException {
    constructor(opts) {
        super({
            name: "CustomKeyStoreInvalidStateException",
            $fault: "client",
            ...opts,
        });
        this.name = "CustomKeyStoreInvalidStateException";
        this.$fault = "client";
        Object.setPrototypeOf(this, CustomKeyStoreInvalidStateException.prototype);
    }
}
exports.CustomKeyStoreInvalidStateException = CustomKeyStoreInvalidStateException;
class CustomKeyStoreNotFoundException extends KMSServiceException_1.KMSServiceException {
    constructor(opts) {
        super({
            name: "CustomKeyStoreNotFoundException",
            $fault: "client",
            ...opts,
        });
        this.name = "CustomKeyStoreNotFoundException";
        this.$fault = "client";
        Object.setPrototypeOf(this, CustomKeyStoreNotFoundException.prototype);
    }
}
exports.CustomKeyStoreNotFoundException = CustomKeyStoreNotFoundException;
var ConnectionErrorCodeType;
(function (ConnectionErrorCodeType) {
    ConnectionErrorCodeType["CLUSTER_NOT_FOUND"] = "CLUSTER_NOT_FOUND";
    ConnectionErrorCodeType["INSUFFICIENT_CLOUDHSM_HSMS"] = "INSUFFICIENT_CLOUDHSM_HSMS";
    ConnectionErrorCodeType["INTERNAL_ERROR"] = "INTERNAL_ERROR";
    ConnectionErrorCodeType["INVALID_CREDENTIALS"] = "INVALID_CREDENTIALS";
    ConnectionErrorCodeType["NETWORK_ERRORS"] = "NETWORK_ERRORS";
    ConnectionErrorCodeType["SUBNET_NOT_FOUND"] = "SUBNET_NOT_FOUND";
    ConnectionErrorCodeType["USER_LOCKED_OUT"] = "USER_LOCKED_OUT";
    ConnectionErrorCodeType["USER_LOGGED_IN"] = "USER_LOGGED_IN";
    ConnectionErrorCodeType["USER_NOT_FOUND"] = "USER_NOT_FOUND";
})(ConnectionErrorCodeType = exports.ConnectionErrorCodeType || (exports.ConnectionErrorCodeType = {}));
var ConnectionStateType;
(function (ConnectionStateType) {
    ConnectionStateType["CONNECTED"] = "CONNECTED";
    ConnectionStateType["CONNECTING"] = "CONNECTING";
    ConnectionStateType["DISCONNECTED"] = "DISCONNECTED";
    ConnectionStateType["DISCONNECTING"] = "DISCONNECTING";
    ConnectionStateType["FAILED"] = "FAILED";
})(ConnectionStateType = exports.ConnectionStateType || (exports.ConnectionStateType = {}));
var CreateAliasRequest;
(function (CreateAliasRequest) {
    CreateAliasRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateAliasRequest = exports.CreateAliasRequest || (exports.CreateAliasRequest = {}));
class InvalidAliasNameException extends KMSServiceException_1.KMSServiceException {
    constructor(opts) {
        super({
            name: "InvalidAliasNameException",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidAliasNameException";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidAliasNameException.prototype);
    }
}
exports.InvalidAliasNameException = InvalidAliasNameException;
class LimitExceededException extends KMSServiceException_1.KMSServiceException {
    constructor(opts) {
        super({
            name: "LimitExceededException",
            $fault: "client",
            ...opts,
        });
        this.name = "LimitExceededException";
        this.$fault = "client";
        Object.setPrototypeOf(this, LimitExceededException.prototype);
    }
}
exports.LimitExceededException = LimitExceededException;
var CreateCustomKeyStoreRequest;
(function (CreateCustomKeyStoreRequest) {
    CreateCustomKeyStoreRequest.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.KeyStorePassword && { KeyStorePassword: smithy_client_1.SENSITIVE_STRING }),
    });
})(CreateCustomKeyStoreRequest = exports.CreateCustomKeyStoreRequest || (exports.CreateCustomKeyStoreRequest = {}));
var CreateCustomKeyStoreResponse;
(function (CreateCustomKeyStoreResponse) {
    CreateCustomKeyStoreResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateCustomKeyStoreResponse = exports.CreateCustomKeyStoreResponse || (exports.CreateCustomKeyStoreResponse = {}));
class CustomKeyStoreNameInUseException extends KMSServiceException_1.KMSServiceException {
    constructor(opts) {
        super({
            name: "CustomKeyStoreNameInUseException",
            $fault: "client",
            ...opts,
        });
        this.name = "CustomKeyStoreNameInUseException";
        this.$fault = "client";
        Object.setPrototypeOf(this, CustomKeyStoreNameInUseException.prototype);
    }
}
exports.CustomKeyStoreNameInUseException = CustomKeyStoreNameInUseException;
class IncorrectTrustAnchorException extends KMSServiceException_1.KMSServiceException {
    constructor(opts) {
        super({
            name: "IncorrectTrustAnchorException",
            $fault: "client",
            ...opts,
        });
        this.name = "IncorrectTrustAnchorException";
        this.$fault = "client";
        Object.setPrototypeOf(this, IncorrectTrustAnchorException.prototype);
    }
}
exports.IncorrectTrustAnchorException = IncorrectTrustAnchorException;
var GrantConstraints;
(function (GrantConstraints) {
    GrantConstraints.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GrantConstraints = exports.GrantConstraints || (exports.GrantConstraints = {}));
var GrantOperation;
(function (GrantOperation) {
    GrantOperation["CreateGrant"] = "CreateGrant";
    GrantOperation["Decrypt"] = "Decrypt";
    GrantOperation["DescribeKey"] = "DescribeKey";
    GrantOperation["Encrypt"] = "Encrypt";
    GrantOperation["GenerateDataKey"] = "GenerateDataKey";
    GrantOperation["GenerateDataKeyPair"] = "GenerateDataKeyPair";
    GrantOperation["GenerateDataKeyPairWithoutPlaintext"] = "GenerateDataKeyPairWithoutPlaintext";
    GrantOperation["GenerateDataKeyWithoutPlaintext"] = "GenerateDataKeyWithoutPlaintext";
    GrantOperation["GetPublicKey"] = "GetPublicKey";
    GrantOperation["ReEncryptFrom"] = "ReEncryptFrom";
    GrantOperation["ReEncryptTo"] = "ReEncryptTo";
    GrantOperation["RetireGrant"] = "RetireGrant";
    GrantOperation["Sign"] = "Sign";
    GrantOperation["Verify"] = "Verify";
})(GrantOperation = exports.GrantOperation || (exports.GrantOperation = {}));
var CreateGrantRequest;
(function (CreateGrantRequest) {
    CreateGrantRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateGrantRequest = exports.CreateGrantRequest || (exports.CreateGrantRequest = {}));
var CreateGrantResponse;
(function (CreateGrantResponse) {
    CreateGrantResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateGrantResponse = exports.CreateGrantResponse || (exports.CreateGrantResponse = {}));
class DisabledException extends KMSServiceException_1.KMSServiceException {
    constructor(opts) {
        super({
            name: "DisabledException",
            $fault: "client",
            ...opts,
        });
        this.name = "DisabledException";
        this.$fault = "client";
        Object.setPrototypeOf(this, DisabledException.prototype);
    }
}
exports.DisabledException = DisabledException;
class InvalidGrantTokenException extends KMSServiceException_1.KMSServiceException {
    constructor(opts) {
        super({
            name: "InvalidGrantTokenException",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidGrantTokenException";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidGrantTokenException.prototype);
    }
}
exports.InvalidGrantTokenException = InvalidGrantTokenException;
var CustomerMasterKeySpec;
(function (CustomerMasterKeySpec) {
    CustomerMasterKeySpec["ECC_NIST_P256"] = "ECC_NIST_P256";
    CustomerMasterKeySpec["ECC_NIST_P384"] = "ECC_NIST_P384";
    CustomerMasterKeySpec["ECC_NIST_P521"] = "ECC_NIST_P521";
    CustomerMasterKeySpec["ECC_SECG_P256K1"] = "ECC_SECG_P256K1";
    CustomerMasterKeySpec["RSA_2048"] = "RSA_2048";
    CustomerMasterKeySpec["RSA_3072"] = "RSA_3072";
    CustomerMasterKeySpec["RSA_4096"] = "RSA_4096";
    CustomerMasterKeySpec["SYMMETRIC_DEFAULT"] = "SYMMETRIC_DEFAULT";
})(CustomerMasterKeySpec = exports.CustomerMasterKeySpec || (exports.CustomerMasterKeySpec = {}));
var KeySpec;
(function (KeySpec) {
    KeySpec["ECC_NIST_P256"] = "ECC_NIST_P256";
    KeySpec["ECC_NIST_P384"] = "ECC_NIST_P384";
    KeySpec["ECC_NIST_P521"] = "ECC_NIST_P521";
    KeySpec["ECC_SECG_P256K1"] = "ECC_SECG_P256K1";
    KeySpec["RSA_2048"] = "RSA_2048";
    KeySpec["RSA_3072"] = "RSA_3072";
    KeySpec["RSA_4096"] = "RSA_4096";
    KeySpec["SYMMETRIC_DEFAULT"] = "SYMMETRIC_DEFAULT";
})(KeySpec = exports.KeySpec || (exports.KeySpec = {}));
var KeyUsageType;
(function (KeyUsageType) {
    KeyUsageType["ENCRYPT_DECRYPT"] = "ENCRYPT_DECRYPT";
    KeyUsageType["SIGN_VERIFY"] = "SIGN_VERIFY";
})(KeyUsageType = exports.KeyUsageType || (exports.KeyUsageType = {}));
var OriginType;
(function (OriginType) {
    OriginType["AWS_CLOUDHSM"] = "AWS_CLOUDHSM";
    OriginType["AWS_KMS"] = "AWS_KMS";
    OriginType["EXTERNAL"] = "EXTERNAL";
})(OriginType = exports.OriginType || (exports.OriginType = {}));
var Tag;
(function (Tag) {
    Tag.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Tag = exports.Tag || (exports.Tag = {}));
var CreateKeyRequest;
(function (CreateKeyRequest) {
    CreateKeyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateKeyRequest = exports.CreateKeyRequest || (exports.CreateKeyRequest = {}));
var EncryptionAlgorithmSpec;
(function (EncryptionAlgorithmSpec) {
    EncryptionAlgorithmSpec["RSAES_OAEP_SHA_1"] = "RSAES_OAEP_SHA_1";
    EncryptionAlgorithmSpec["RSAES_OAEP_SHA_256"] = "RSAES_OAEP_SHA_256";
    EncryptionAlgorithmSpec["SYMMETRIC_DEFAULT"] = "SYMMETRIC_DEFAULT";
})(EncryptionAlgorithmSpec = exports.EncryptionAlgorithmSpec || (exports.EncryptionAlgorithmSpec = {}));
var ExpirationModelType;
(function (ExpirationModelType) {
    ExpirationModelType["KEY_MATERIAL_DOES_NOT_EXPIRE"] = "KEY_MATERIAL_DOES_NOT_EXPIRE";
    ExpirationModelType["KEY_MATERIAL_EXPIRES"] = "KEY_MATERIAL_EXPIRES";
})(ExpirationModelType = exports.ExpirationModelType || (exports.ExpirationModelType = {}));
var KeyManagerType;
(function (KeyManagerType) {
    KeyManagerType["AWS"] = "AWS";
    KeyManagerType["CUSTOMER"] = "CUSTOMER";
})(KeyManagerType = exports.KeyManagerType || (exports.KeyManagerType = {}));
var KeyState;
(function (KeyState) {
    KeyState["Creating"] = "Creating";
    KeyState["Disabled"] = "Disabled";
    KeyState["Enabled"] = "Enabled";
    KeyState["PendingDeletion"] = "PendingDeletion";
    KeyState["PendingImport"] = "PendingImport";
    KeyState["PendingReplicaDeletion"] = "PendingReplicaDeletion";
    KeyState["Unavailable"] = "Unavailable";
    KeyState["Updating"] = "Updating";
})(KeyState = exports.KeyState || (exports.KeyState = {}));
var MultiRegionKeyType;
(function (MultiRegionKeyType) {
    MultiRegionKeyType["PRIMARY"] = "PRIMARY";
    MultiRegionKeyType["REPLICA"] = "REPLICA";
})(MultiRegionKeyType = exports.MultiRegionKeyType || (exports.MultiRegionKeyType = {}));
var MultiRegionKey;
(function (MultiRegionKey) {
    MultiRegionKey.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(MultiRegionKey = exports.MultiRegionKey || (exports.MultiRegionKey = {}));
var MultiRegionConfiguration;
(function (MultiRegionConfiguration) {
    MultiRegionConfiguration.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(MultiRegionConfiguration = exports.MultiRegionConfiguration || (exports.MultiRegionConfiguration = {}));
var SigningAlgorithmSpec;
(function (SigningAlgorithmSpec) {
    SigningAlgorithmSpec["ECDSA_SHA_256"] = "ECDSA_SHA_256";
    SigningAlgorithmSpec["ECDSA_SHA_384"] = "ECDSA_SHA_384";
    SigningAlgorithmSpec["ECDSA_SHA_512"] = "ECDSA_SHA_512";
    SigningAlgorithmSpec["RSASSA_PKCS1_V1_5_SHA_256"] = "RSASSA_PKCS1_V1_5_SHA_256";
    SigningAlgorithmSpec["RSASSA_PKCS1_V1_5_SHA_384"] = "RSASSA_PKCS1_V1_5_SHA_384";
    SigningAlgorithmSpec["RSASSA_PKCS1_V1_5_SHA_512"] = "RSASSA_PKCS1_V1_5_SHA_512";
    SigningAlgorithmSpec["RSASSA_PSS_SHA_256"] = "RSASSA_PSS_SHA_256";
    SigningAlgorithmSpec["RSASSA_PSS_SHA_384"] = "RSASSA_PSS_SHA_384";
    SigningAlgorithmSpec["RSASSA_PSS_SHA_512"] = "RSASSA_PSS_SHA_512";
})(SigningAlgorithmSpec = exports.SigningAlgorithmSpec || (exports.SigningAlgorithmSpec = {}));
var KeyMetadata;
(function (KeyMetadata) {
    KeyMetadata.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(KeyMetadata = exports.KeyMetadata || (exports.KeyMetadata = {}));
var CreateKeyResponse;
(function (CreateKeyResponse) {
    CreateKeyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateKeyResponse = exports.CreateKeyResponse || (exports.CreateKeyResponse = {}));
class MalformedPolicyDocumentException extends KMSServiceException_1.KMSServiceException {
    constructor(opts) {
        super({
            name: "MalformedPolicyDocumentException",
            $fault: "client",
            ...opts,
        });
        this.name = "MalformedPolicyDocumentException";
        this.$fault = "client";
        Object.setPrototypeOf(this, MalformedPolicyDocumentException.prototype);
    }
}
exports.MalformedPolicyDocumentException = MalformedPolicyDocumentException;
class TagException extends KMSServiceException_1.KMSServiceException {
    constructor(opts) {
        super({
            name: "TagException",
            $fault: "client",
            ...opts,
        });
        this.name = "TagException";
        this.$fault = "client";
        Object.setPrototypeOf(this, TagException.prototype);
    }
}
exports.TagException = TagException;
class UnsupportedOperationException extends KMSServiceException_1.KMSServiceException {
    constructor(opts) {
        super({
            name: "UnsupportedOperationException",
            $fault: "client",
            ...opts,
        });
        this.name = "UnsupportedOperationException";
        this.$fault = "client";
        Object.setPrototypeOf(this, UnsupportedOperationException.prototype);
    }
}
exports.UnsupportedOperationException = UnsupportedOperationException;
class CustomKeyStoreHasCMKsException extends KMSServiceException_1.KMSServiceException {
    constructor(opts) {
        super({
            name: "CustomKeyStoreHasCMKsException",
            $fault: "client",
            ...opts,
        });
        this.name = "CustomKeyStoreHasCMKsException";
        this.$fault = "client";
        Object.setPrototypeOf(this, CustomKeyStoreHasCMKsException.prototype);
    }
}
exports.CustomKeyStoreHasCMKsException = CustomKeyStoreHasCMKsException;
var CustomKeyStoresListEntry;
(function (CustomKeyStoresListEntry) {
    CustomKeyStoresListEntry.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CustomKeyStoresListEntry = exports.CustomKeyStoresListEntry || (exports.CustomKeyStoresListEntry = {}));
var DataKeyPairSpec;
(function (DataKeyPairSpec) {
    DataKeyPairSpec["ECC_NIST_P256"] = "ECC_NIST_P256";
    DataKeyPairSpec["ECC_NIST_P384"] = "ECC_NIST_P384";
    DataKeyPairSpec["ECC_NIST_P521"] = "ECC_NIST_P521";
    DataKeyPairSpec["ECC_SECG_P256K1"] = "ECC_SECG_P256K1";
    DataKeyPairSpec["RSA_2048"] = "RSA_2048";
    DataKeyPairSpec["RSA_3072"] = "RSA_3072";
    DataKeyPairSpec["RSA_4096"] = "RSA_4096";
})(DataKeyPairSpec = exports.DataKeyPairSpec || (exports.DataKeyPairSpec = {}));
var DataKeySpec;
(function (DataKeySpec) {
    DataKeySpec["AES_128"] = "AES_128";
    DataKeySpec["AES_256"] = "AES_256";
})(DataKeySpec = exports.DataKeySpec || (exports.DataKeySpec = {}));
var DecryptRequest;
(function (DecryptRequest) {
    DecryptRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DecryptRequest = exports.DecryptRequest || (exports.DecryptRequest = {}));
var DecryptResponse;
(function (DecryptResponse) {
    DecryptResponse.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.Plaintext && { Plaintext: smithy_client_1.SENSITIVE_STRING }),
    });
})(DecryptResponse = exports.DecryptResponse || (exports.DecryptResponse = {}));
class IncorrectKeyException extends KMSServiceException_1.KMSServiceException {
    constructor(opts) {
        super({
            name: "IncorrectKeyException",
            $fault: "client",
            ...opts,
        });
        this.name = "IncorrectKeyException";
        this.$fault = "client";
        Object.setPrototypeOf(this, IncorrectKeyException.prototype);
    }
}
exports.IncorrectKeyException = IncorrectKeyException;
class InvalidCiphertextException extends KMSServiceException_1.KMSServiceException {
    constructor(opts) {
        super({
            name: "InvalidCiphertextException",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidCiphertextException";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidCiphertextException.prototype);
    }
}
exports.InvalidCiphertextException = InvalidCiphertextException;
class InvalidKeyUsageException extends KMSServiceException_1.KMSServiceException {
    constructor(opts) {
        super({
            name: "InvalidKeyUsageException",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidKeyUsageException";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidKeyUsageException.prototype);
    }
}
exports.InvalidKeyUsageException = InvalidKeyUsageException;
class KeyUnavailableException extends KMSServiceException_1.KMSServiceException {
    constructor(opts) {
        super({
            name: "KeyUnavailableException",
            $fault: "server",
            ...opts,
        });
        this.name = "KeyUnavailableException";
        this.$fault = "server";
        Object.setPrototypeOf(this, KeyUnavailableException.prototype);
    }
}
exports.KeyUnavailableException = KeyUnavailableException;
var DeleteAliasRequest;
(function (DeleteAliasRequest) {
    DeleteAliasRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteAliasRequest = exports.DeleteAliasRequest || (exports.DeleteAliasRequest = {}));
var DeleteCustomKeyStoreRequest;
(function (DeleteCustomKeyStoreRequest) {
    DeleteCustomKeyStoreRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteCustomKeyStoreRequest = exports.DeleteCustomKeyStoreRequest || (exports.DeleteCustomKeyStoreRequest = {}));
var DeleteCustomKeyStoreResponse;
(function (DeleteCustomKeyStoreResponse) {
    DeleteCustomKeyStoreResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteCustomKeyStoreResponse = exports.DeleteCustomKeyStoreResponse || (exports.DeleteCustomKeyStoreResponse = {}));
var DeleteImportedKeyMaterialRequest;
(function (DeleteImportedKeyMaterialRequest) {
    DeleteImportedKeyMaterialRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteImportedKeyMaterialRequest = exports.DeleteImportedKeyMaterialRequest || (exports.DeleteImportedKeyMaterialRequest = {}));
var DescribeCustomKeyStoresRequest;
(function (DescribeCustomKeyStoresRequest) {
    DescribeCustomKeyStoresRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeCustomKeyStoresRequest = exports.DescribeCustomKeyStoresRequest || (exports.DescribeCustomKeyStoresRequest = {}));
var DescribeCustomKeyStoresResponse;
(function (DescribeCustomKeyStoresResponse) {
    DescribeCustomKeyStoresResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeCustomKeyStoresResponse = exports.DescribeCustomKeyStoresResponse || (exports.DescribeCustomKeyStoresResponse = {}));
class InvalidMarkerException extends KMSServiceException_1.KMSServiceException {
    constructor(opts) {
        super({
            name: "InvalidMarkerException",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidMarkerException";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidMarkerException.prototype);
    }
}
exports.InvalidMarkerException = InvalidMarkerException;
var DescribeKeyRequest;
(function (DescribeKeyRequest) {
    DescribeKeyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeKeyRequest = exports.DescribeKeyRequest || (exports.DescribeKeyRequest = {}));
var DescribeKeyResponse;
(function (DescribeKeyResponse) {
    DescribeKeyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeKeyResponse = exports.DescribeKeyResponse || (exports.DescribeKeyResponse = {}));
var DisableKeyRequest;
(function (DisableKeyRequest) {
    DisableKeyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DisableKeyRequest = exports.DisableKeyRequest || (exports.DisableKeyRequest = {}));
var DisableKeyRotationRequest;
(function (DisableKeyRotationRequest) {
    DisableKeyRotationRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DisableKeyRotationRequest = exports.DisableKeyRotationRequest || (exports.DisableKeyRotationRequest = {}));
var DisconnectCustomKeyStoreRequest;
(function (DisconnectCustomKeyStoreRequest) {
    DisconnectCustomKeyStoreRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DisconnectCustomKeyStoreRequest = exports.DisconnectCustomKeyStoreRequest || (exports.DisconnectCustomKeyStoreRequest = {}));
var DisconnectCustomKeyStoreResponse;
(function (DisconnectCustomKeyStoreResponse) {
    DisconnectCustomKeyStoreResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DisconnectCustomKeyStoreResponse = exports.DisconnectCustomKeyStoreResponse || (exports.DisconnectCustomKeyStoreResponse = {}));
var EnableKeyRequest;
(function (EnableKeyRequest) {
    EnableKeyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(EnableKeyRequest = exports.EnableKeyRequest || (exports.EnableKeyRequest = {}));
var EnableKeyRotationRequest;
(function (EnableKeyRotationRequest) {
    EnableKeyRotationRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(EnableKeyRotationRequest = exports.EnableKeyRotationRequest || (exports.EnableKeyRotationRequest = {}));
var EncryptRequest;
(function (EncryptRequest) {
    EncryptRequest.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.Plaintext && { Plaintext: smithy_client_1.SENSITIVE_STRING }),
    });
})(EncryptRequest = exports.EncryptRequest || (exports.EncryptRequest = {}));
var EncryptResponse;
(function (EncryptResponse) {
    EncryptResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(EncryptResponse = exports.EncryptResponse || (exports.EncryptResponse = {}));
class ExpiredImportTokenException extends KMSServiceException_1.KMSServiceException {
    constructor(opts) {
        super({
            name: "ExpiredImportTokenException",
            $fault: "client",
            ...opts,
        });
        this.name = "ExpiredImportTokenException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ExpiredImportTokenException.prototype);
    }
}
exports.ExpiredImportTokenException = ExpiredImportTokenException;
var GenerateDataKeyRequest;
(function (GenerateDataKeyRequest) {
    GenerateDataKeyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GenerateDataKeyRequest = exports.GenerateDataKeyRequest || (exports.GenerateDataKeyRequest = {}));
var GenerateDataKeyResponse;
(function (GenerateDataKeyResponse) {
    GenerateDataKeyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.Plaintext && { Plaintext: smithy_client_1.SENSITIVE_STRING }),
    });
})(GenerateDataKeyResponse = exports.GenerateDataKeyResponse || (exports.GenerateDataKeyResponse = {}));
var GenerateDataKeyPairRequest;
(function (GenerateDataKeyPairRequest) {
    GenerateDataKeyPairRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GenerateDataKeyPairRequest = exports.GenerateDataKeyPairRequest || (exports.GenerateDataKeyPairRequest = {}));
var GenerateDataKeyPairResponse;
(function (GenerateDataKeyPairResponse) {
    GenerateDataKeyPairResponse.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.PrivateKeyPlaintext && { PrivateKeyPlaintext: smithy_client_1.SENSITIVE_STRING }),
    });
})(GenerateDataKeyPairResponse = exports.GenerateDataKeyPairResponse || (exports.GenerateDataKeyPairResponse = {}));
var GenerateDataKeyPairWithoutPlaintextRequest;
(function (GenerateDataKeyPairWithoutPlaintextRequest) {
    GenerateDataKeyPairWithoutPlaintextRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GenerateDataKeyPairWithoutPlaintextRequest = exports.GenerateDataKeyPairWithoutPlaintextRequest || (exports.GenerateDataKeyPairWithoutPlaintextRequest = {}));
var GenerateDataKeyPairWithoutPlaintextResponse;
(function (GenerateDataKeyPairWithoutPlaintextResponse) {
    GenerateDataKeyPairWithoutPlaintextResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GenerateDataKeyPairWithoutPlaintextResponse = exports.GenerateDataKeyPairWithoutPlaintextResponse || (exports.GenerateDataKeyPairWithoutPlaintextResponse = {}));
var GenerateDataKeyWithoutPlaintextRequest;
(function (GenerateDataKeyWithoutPlaintextRequest) {
    GenerateDataKeyWithoutPlaintextRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GenerateDataKeyWithoutPlaintextRequest = exports.GenerateDataKeyWithoutPlaintextRequest || (exports.GenerateDataKeyWithoutPlaintextRequest = {}));
var GenerateDataKeyWithoutPlaintextResponse;
(function (GenerateDataKeyWithoutPlaintextResponse) {
    GenerateDataKeyWithoutPlaintextResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GenerateDataKeyWithoutPlaintextResponse = exports.GenerateDataKeyWithoutPlaintextResponse || (exports.GenerateDataKeyWithoutPlaintextResponse = {}));
var GenerateRandomRequest;
(function (GenerateRandomRequest) {
    GenerateRandomRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GenerateRandomRequest = exports.GenerateRandomRequest || (exports.GenerateRandomRequest = {}));
var GenerateRandomResponse;
(function (GenerateRandomResponse) {
    GenerateRandomResponse.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.Plaintext && { Plaintext: smithy_client_1.SENSITIVE_STRING }),
    });
})(GenerateRandomResponse = exports.GenerateRandomResponse || (exports.GenerateRandomResponse = {}));
var GetKeyPolicyRequest;
(function (GetKeyPolicyRequest) {
    GetKeyPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetKeyPolicyRequest = exports.GetKeyPolicyRequest || (exports.GetKeyPolicyRequest = {}));
var GetKeyPolicyResponse;
(function (GetKeyPolicyResponse) {
    GetKeyPolicyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetKeyPolicyResponse = exports.GetKeyPolicyResponse || (exports.GetKeyPolicyResponse = {}));
var GetKeyRotationStatusRequest;
(function (GetKeyRotationStatusRequest) {
    GetKeyRotationStatusRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetKeyRotationStatusRequest = exports.GetKeyRotationStatusRequest || (exports.GetKeyRotationStatusRequest = {}));
var GetKeyRotationStatusResponse;
(function (GetKeyRotationStatusResponse) {
    GetKeyRotationStatusResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetKeyRotationStatusResponse = exports.GetKeyRotationStatusResponse || (exports.GetKeyRotationStatusResponse = {}));
var WrappingKeySpec;
(function (WrappingKeySpec) {
    WrappingKeySpec["RSA_2048"] = "RSA_2048";
})(WrappingKeySpec = exports.WrappingKeySpec || (exports.WrappingKeySpec = {}));
var GetParametersForImportRequest;
(function (GetParametersForImportRequest) {
    GetParametersForImportRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetParametersForImportRequest = exports.GetParametersForImportRequest || (exports.GetParametersForImportRequest = {}));
var GetParametersForImportResponse;
(function (GetParametersForImportResponse) {
    GetParametersForImportResponse.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.PublicKey && { PublicKey: smithy_client_1.SENSITIVE_STRING }),
    });
})(GetParametersForImportResponse = exports.GetParametersForImportResponse || (exports.GetParametersForImportResponse = {}));
var GetPublicKeyRequest;
(function (GetPublicKeyRequest) {
    GetPublicKeyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetPublicKeyRequest = exports.GetPublicKeyRequest || (exports.GetPublicKeyRequest = {}));
var GetPublicKeyResponse;
(function (GetPublicKeyResponse) {
    GetPublicKeyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetPublicKeyResponse = exports.GetPublicKeyResponse || (exports.GetPublicKeyResponse = {}));
var GrantListEntry;
(function (GrantListEntry) {
    GrantListEntry.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GrantListEntry = exports.GrantListEntry || (exports.GrantListEntry = {}));
var ImportKeyMaterialRequest;
(function (ImportKeyMaterialRequest) {
    ImportKeyMaterialRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ImportKeyMaterialRequest = exports.ImportKeyMaterialRequest || (exports.ImportKeyMaterialRequest = {}));
var ImportKeyMaterialResponse;
(function (ImportKeyMaterialResponse) {
    ImportKeyMaterialResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ImportKeyMaterialResponse = exports.ImportKeyMaterialResponse || (exports.ImportKeyMaterialResponse = {}));
class IncorrectKeyMaterialException extends KMSServiceException_1.KMSServiceException {
    constructor(opts) {
        super({
            name: "IncorrectKeyMaterialException",
            $fault: "client",
            ...opts,
        });
        this.name = "IncorrectKeyMaterialException";
        this.$fault = "client";
        Object.setPrototypeOf(this, IncorrectKeyMaterialException.prototype);
    }
}
exports.IncorrectKeyMaterialException = IncorrectKeyMaterialException;
class InvalidImportTokenException extends KMSServiceException_1.KMSServiceException {
    constructor(opts) {
        super({
            name: "InvalidImportTokenException",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidImportTokenException";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidImportTokenException.prototype);
    }
}
exports.InvalidImportTokenException = InvalidImportTokenException;
class InvalidGrantIdException extends KMSServiceException_1.KMSServiceException {
    constructor(opts) {
        super({
            name: "InvalidGrantIdException",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidGrantIdException";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidGrantIdException.prototype);
    }
}
exports.InvalidGrantIdException = InvalidGrantIdException;
var KeyListEntry;
(function (KeyListEntry) {
    KeyListEntry.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(KeyListEntry = exports.KeyListEntry || (exports.KeyListEntry = {}));
class KMSInvalidSignatureException extends KMSServiceException_1.KMSServiceException {
    constructor(opts) {
        super({
            name: "KMSInvalidSignatureException",
            $fault: "client",
            ...opts,
        });
        this.name = "KMSInvalidSignatureException";
        this.$fault = "client";
        Object.setPrototypeOf(this, KMSInvalidSignatureException.prototype);
    }
}
exports.KMSInvalidSignatureException = KMSInvalidSignatureException;
var ListAliasesRequest;
(function (ListAliasesRequest) {
    ListAliasesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListAliasesRequest = exports.ListAliasesRequest || (exports.ListAliasesRequest = {}));
var ListAliasesResponse;
(function (ListAliasesResponse) {
    ListAliasesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListAliasesResponse = exports.ListAliasesResponse || (exports.ListAliasesResponse = {}));
var ListGrantsRequest;
(function (ListGrantsRequest) {
    ListGrantsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListGrantsRequest = exports.ListGrantsRequest || (exports.ListGrantsRequest = {}));
var ListGrantsResponse;
(function (ListGrantsResponse) {
    ListGrantsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListGrantsResponse = exports.ListGrantsResponse || (exports.ListGrantsResponse = {}));
var ListKeyPoliciesRequest;
(function (ListKeyPoliciesRequest) {
    ListKeyPoliciesRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListKeyPoliciesRequest = exports.ListKeyPoliciesRequest || (exports.ListKeyPoliciesRequest = {}));
var ListKeyPoliciesResponse;
(function (ListKeyPoliciesResponse) {
    ListKeyPoliciesResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListKeyPoliciesResponse = exports.ListKeyPoliciesResponse || (exports.ListKeyPoliciesResponse = {}));
var ListKeysRequest;
(function (ListKeysRequest) {
    ListKeysRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListKeysRequest = exports.ListKeysRequest || (exports.ListKeysRequest = {}));
var ListKeysResponse;
(function (ListKeysResponse) {
    ListKeysResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListKeysResponse = exports.ListKeysResponse || (exports.ListKeysResponse = {}));
var ListResourceTagsRequest;
(function (ListResourceTagsRequest) {
    ListResourceTagsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListResourceTagsRequest = exports.ListResourceTagsRequest || (exports.ListResourceTagsRequest = {}));
var ListResourceTagsResponse;
(function (ListResourceTagsResponse) {
    ListResourceTagsResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListResourceTagsResponse = exports.ListResourceTagsResponse || (exports.ListResourceTagsResponse = {}));
var ListRetirableGrantsRequest;
(function (ListRetirableGrantsRequest) {
    ListRetirableGrantsRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListRetirableGrantsRequest = exports.ListRetirableGrantsRequest || (exports.ListRetirableGrantsRequest = {}));
var MessageType;
(function (MessageType) {
    MessageType["DIGEST"] = "DIGEST";
    MessageType["RAW"] = "RAW";
})(MessageType = exports.MessageType || (exports.MessageType = {}));
var PutKeyPolicyRequest;
(function (PutKeyPolicyRequest) {
    PutKeyPolicyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PutKeyPolicyRequest = exports.PutKeyPolicyRequest || (exports.PutKeyPolicyRequest = {}));
var ReEncryptRequest;
(function (ReEncryptRequest) {
    ReEncryptRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ReEncryptRequest = exports.ReEncryptRequest || (exports.ReEncryptRequest = {}));
var ReEncryptResponse;
(function (ReEncryptResponse) {
    ReEncryptResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ReEncryptResponse = exports.ReEncryptResponse || (exports.ReEncryptResponse = {}));
var ReplicateKeyRequest;
(function (ReplicateKeyRequest) {
    ReplicateKeyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ReplicateKeyRequest = exports.ReplicateKeyRequest || (exports.ReplicateKeyRequest = {}));
var ReplicateKeyResponse;
(function (ReplicateKeyResponse) {
    ReplicateKeyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ReplicateKeyResponse = exports.ReplicateKeyResponse || (exports.ReplicateKeyResponse = {}));
var RetireGrantRequest;
(function (RetireGrantRequest) {
    RetireGrantRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RetireGrantRequest = exports.RetireGrantRequest || (exports.RetireGrantRequest = {}));
var RevokeGrantRequest;
(function (RevokeGrantRequest) {
    RevokeGrantRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RevokeGrantRequest = exports.RevokeGrantRequest || (exports.RevokeGrantRequest = {}));
var ScheduleKeyDeletionRequest;
(function (ScheduleKeyDeletionRequest) {
    ScheduleKeyDeletionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ScheduleKeyDeletionRequest = exports.ScheduleKeyDeletionRequest || (exports.ScheduleKeyDeletionRequest = {}));
var ScheduleKeyDeletionResponse;
(function (ScheduleKeyDeletionResponse) {
    ScheduleKeyDeletionResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ScheduleKeyDeletionResponse = exports.ScheduleKeyDeletionResponse || (exports.ScheduleKeyDeletionResponse = {}));
var SignRequest;
(function (SignRequest) {
    SignRequest.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.Message && { Message: smithy_client_1.SENSITIVE_STRING }),
    });
})(SignRequest = exports.SignRequest || (exports.SignRequest = {}));
var SignResponse;
(function (SignResponse) {
    SignResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SignResponse = exports.SignResponse || (exports.SignResponse = {}));
var TagResourceRequest;
(function (TagResourceRequest) {
    TagResourceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TagResourceRequest = exports.TagResourceRequest || (exports.TagResourceRequest = {}));
var UntagResourceRequest;
(function (UntagResourceRequest) {
    UntagResourceRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UntagResourceRequest = exports.UntagResourceRequest || (exports.UntagResourceRequest = {}));
var UpdateAliasRequest;
(function (UpdateAliasRequest) {
    UpdateAliasRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateAliasRequest = exports.UpdateAliasRequest || (exports.UpdateAliasRequest = {}));
var UpdateCustomKeyStoreRequest;
(function (UpdateCustomKeyStoreRequest) {
    UpdateCustomKeyStoreRequest.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.KeyStorePassword && { KeyStorePassword: smithy_client_1.SENSITIVE_STRING }),
    });
})(UpdateCustomKeyStoreRequest = exports.UpdateCustomKeyStoreRequest || (exports.UpdateCustomKeyStoreRequest = {}));
var UpdateCustomKeyStoreResponse;
(function (UpdateCustomKeyStoreResponse) {
    UpdateCustomKeyStoreResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateCustomKeyStoreResponse = exports.UpdateCustomKeyStoreResponse || (exports.UpdateCustomKeyStoreResponse = {}));
var UpdateKeyDescriptionRequest;
(function (UpdateKeyDescriptionRequest) {
    UpdateKeyDescriptionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateKeyDescriptionRequest = exports.UpdateKeyDescriptionRequest || (exports.UpdateKeyDescriptionRequest = {}));
var UpdatePrimaryRegionRequest;
(function (UpdatePrimaryRegionRequest) {
    UpdatePrimaryRegionRequest.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdatePrimaryRegionRequest = exports.UpdatePrimaryRegionRequest || (exports.UpdatePrimaryRegionRequest = {}));
var VerifyRequest;
(function (VerifyRequest) {
    VerifyRequest.filterSensitiveLog = (obj) => ({
        ...obj,
        ...(obj.Message && { Message: smithy_client_1.SENSITIVE_STRING }),
    });
})(VerifyRequest = exports.VerifyRequest || (exports.VerifyRequest = {}));
var VerifyResponse;
(function (VerifyResponse) {
    VerifyResponse.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(VerifyResponse = exports.VerifyResponse || (exports.VerifyResponse = {}));
