// @aws-sdk/client-kms/dist-es/models/models_0.js
import { __assign, __extends } from "tslib";
import { SENSITIVE_STRING } from "@aws-sdk/smithy-client";
import { KMSServiceException as __BaseException } from "./KMSServiceException";
export var AlgorithmSpec;
(function (AlgorithmSpec) {
    AlgorithmSpec["RSAES_OAEP_SHA_1"] = "RSAES_OAEP_SHA_1";
    AlgorithmSpec["RSAES_OAEP_SHA_256"] = "RSAES_OAEP_SHA_256";
    AlgorithmSpec["RSAES_PKCS1_V1_5"] = "RSAES_PKCS1_V1_5";
})(AlgorithmSpec || (AlgorithmSpec = {}));
export var AliasListEntry;
(function (AliasListEntry) {
    AliasListEntry.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AliasListEntry || (AliasListEntry = {}));
var AlreadyExistsException = (function (_super) {
    __extends(AlreadyExistsException, _super);
    function AlreadyExistsException(opts) {
        var _this = _super.call(this, __assign({ name: "AlreadyExistsException", $fault: "client" }, opts)) || this;
        _this.name = "AlreadyExistsException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, AlreadyExistsException.prototype);
        return _this;
    }
    return AlreadyExistsException;
}(__BaseException));
export { AlreadyExistsException };
export var CancelKeyDeletionRequest;
(function (CancelKeyDeletionRequest) {
    CancelKeyDeletionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CancelKeyDeletionRequest || (CancelKeyDeletionRequest = {}));
export var CancelKeyDeletionResponse;
(function (CancelKeyDeletionResponse) {
    CancelKeyDeletionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CancelKeyDeletionResponse || (CancelKeyDeletionResponse = {}));
var DependencyTimeoutException = (function (_super) {
    __extends(DependencyTimeoutException, _super);
    function DependencyTimeoutException(opts) {
        var _this = _super.call(this, __assign({ name: "DependencyTimeoutException", $fault: "server" }, opts)) || this;
        _this.name = "DependencyTimeoutException";
        _this.$fault = "server";
        Object.setPrototypeOf(_this, DependencyTimeoutException.prototype);
        return _this;
    }
    return DependencyTimeoutException;
}(__BaseException));
export { DependencyTimeoutException };
var InvalidArnException = (function (_super) {
    __extends(InvalidArnException, _super);
    function InvalidArnException(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidArnException", $fault: "client" }, opts)) || this;
        _this.name = "InvalidArnException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidArnException.prototype);
        return _this;
    }
    return InvalidArnException;
}(__BaseException));
export { InvalidArnException };
var KMSInternalException = (function (_super) {
    __extends(KMSInternalException, _super);
    function KMSInternalException(opts) {
        var _this = _super.call(this, __assign({ name: "KMSInternalException", $fault: "server" }, opts)) || this;
        _this.name = "KMSInternalException";
        _this.$fault = "server";
        Object.setPrototypeOf(_this, KMSInternalException.prototype);
        return _this;
    }
    return KMSInternalException;
}(__BaseException));
export { KMSInternalException };
var KMSInvalidStateException = (function (_super) {
    __extends(KMSInvalidStateException, _super);
    function KMSInvalidStateException(opts) {
        var _this = _super.call(this, __assign({ name: "KMSInvalidStateException", $fault: "client" }, opts)) || this;
        _this.name = "KMSInvalidStateException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, KMSInvalidStateException.prototype);
        return _this;
    }
    return KMSInvalidStateException;
}(__BaseException));
export { KMSInvalidStateException };
var NotFoundException = (function (_super) {
    __extends(NotFoundException, _super);
    function NotFoundException(opts) {
        var _this = _super.call(this, __assign({ name: "NotFoundException", $fault: "client" }, opts)) || this;
        _this.name = "NotFoundException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, NotFoundException.prototype);
        return _this;
    }
    return NotFoundException;
}(__BaseException));
export { NotFoundException };
var CloudHsmClusterInUseException = (function (_super) {
    __extends(CloudHsmClusterInUseException, _super);
    function CloudHsmClusterInUseException(opts) {
        var _this = _super.call(this, __assign({ name: "CloudHsmClusterInUseException", $fault: "client" }, opts)) || this;
        _this.name = "CloudHsmClusterInUseException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, CloudHsmClusterInUseException.prototype);
        return _this;
    }
    return CloudHsmClusterInUseException;
}(__BaseException));
export { CloudHsmClusterInUseException };
var CloudHsmClusterInvalidConfigurationException = (function (_super) {
    __extends(CloudHsmClusterInvalidConfigurationException, _super);
    function CloudHsmClusterInvalidConfigurationException(opts) {
        var _this = _super.call(this, __assign({ name: "CloudHsmClusterInvalidConfigurationException", $fault: "client" }, opts)) || this;
        _this.name = "CloudHsmClusterInvalidConfigurationException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, CloudHsmClusterInvalidConfigurationException.prototype);
        return _this;
    }
    return CloudHsmClusterInvalidConfigurationException;
}(__BaseException));
export { CloudHsmClusterInvalidConfigurationException };
var CloudHsmClusterNotActiveException = (function (_super) {
    __extends(CloudHsmClusterNotActiveException, _super);
    function CloudHsmClusterNotActiveException(opts) {
        var _this = _super.call(this, __assign({ name: "CloudHsmClusterNotActiveException", $fault: "client" }, opts)) || this;
        _this.name = "CloudHsmClusterNotActiveException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, CloudHsmClusterNotActiveException.prototype);
        return _this;
    }
    return CloudHsmClusterNotActiveException;
}(__BaseException));
export { CloudHsmClusterNotActiveException };
var CloudHsmClusterNotFoundException = (function (_super) {
    __extends(CloudHsmClusterNotFoundException, _super);
    function CloudHsmClusterNotFoundException(opts) {
        var _this = _super.call(this, __assign({ name: "CloudHsmClusterNotFoundException", $fault: "client" }, opts)) || this;
        _this.name = "CloudHsmClusterNotFoundException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, CloudHsmClusterNotFoundException.prototype);
        return _this;
    }
    return CloudHsmClusterNotFoundException;
}(__BaseException));
export { CloudHsmClusterNotFoundException };
var CloudHsmClusterNotRelatedException = (function (_super) {
    __extends(CloudHsmClusterNotRelatedException, _super);
    function CloudHsmClusterNotRelatedException(opts) {
        var _this = _super.call(this, __assign({ name: "CloudHsmClusterNotRelatedException", $fault: "client" }, opts)) || this;
        _this.name = "CloudHsmClusterNotRelatedException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, CloudHsmClusterNotRelatedException.prototype);
        return _this;
    }
    return CloudHsmClusterNotRelatedException;
}(__BaseException));
export { CloudHsmClusterNotRelatedException };
export var ConnectCustomKeyStoreRequest;
(function (ConnectCustomKeyStoreRequest) {
    ConnectCustomKeyStoreRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ConnectCustomKeyStoreRequest || (ConnectCustomKeyStoreRequest = {}));
export var ConnectCustomKeyStoreResponse;
(function (ConnectCustomKeyStoreResponse) {
    ConnectCustomKeyStoreResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ConnectCustomKeyStoreResponse || (ConnectCustomKeyStoreResponse = {}));
var CustomKeyStoreInvalidStateException = (function (_super) {
    __extends(CustomKeyStoreInvalidStateException, _super);
    function CustomKeyStoreInvalidStateException(opts) {
        var _this = _super.call(this, __assign({ name: "CustomKeyStoreInvalidStateException", $fault: "client" }, opts)) || this;
        _this.name = "CustomKeyStoreInvalidStateException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, CustomKeyStoreInvalidStateException.prototype);
        return _this;
    }
    return CustomKeyStoreInvalidStateException;
}(__BaseException));
export { CustomKeyStoreInvalidStateException };
var CustomKeyStoreNotFoundException = (function (_super) {
    __extends(CustomKeyStoreNotFoundException, _super);
    function CustomKeyStoreNotFoundException(opts) {
        var _this = _super.call(this, __assign({ name: "CustomKeyStoreNotFoundException", $fault: "client" }, opts)) || this;
        _this.name = "CustomKeyStoreNotFoundException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, CustomKeyStoreNotFoundException.prototype);
        return _this;
    }
    return CustomKeyStoreNotFoundException;
}(__BaseException));
export { CustomKeyStoreNotFoundException };
export var ConnectionErrorCodeType;
(function (ConnectionErrorCodeType) {
    ConnectionErrorCodeType["CLUSTER_NOT_FOUND"] = "CLUSTER_NOT_FOUND";
    ConnectionErrorCodeType["INSUFFICIENT_CLOUDHSM_HSMS"] = "INSUFFICIENT_CLOUDHSM_HSMS";
    ConnectionErrorCodeType["INTERNAL_ERROR"] = "INTERNAL_ERROR";
    ConnectionErrorCodeType["INVALID_CREDENTIALS"] = "INVALID_CREDENTIALS";
    ConnectionErrorCodeType["NETWORK_ERRORS"] = "NETWORK_ERRORS";
    ConnectionErrorCodeType["SUBNET_NOT_FOUND"] = "SUBNET_NOT_FOUND";
    ConnectionErrorCodeType["USER_LOCKED_OUT"] = "USER_LOCKED_OUT";
    ConnectionErrorCodeType["USER_LOGGED_IN"] = "USER_LOGGED_IN";
    ConnectionErrorCodeType["USER_NOT_FOUND"] = "USER_NOT_FOUND";
})(ConnectionErrorCodeType || (ConnectionErrorCodeType = {}));
export var ConnectionStateType;
(function (ConnectionStateType) {
    ConnectionStateType["CONNECTED"] = "CONNECTED";
    ConnectionStateType["CONNECTING"] = "CONNECTING";
    ConnectionStateType["DISCONNECTED"] = "DISCONNECTED";
    ConnectionStateType["DISCONNECTING"] = "DISCONNECTING";
    ConnectionStateType["FAILED"] = "FAILED";
})(ConnectionStateType || (ConnectionStateType = {}));
export var CreateAliasRequest;
(function (CreateAliasRequest) {
    CreateAliasRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateAliasRequest || (CreateAliasRequest = {}));
var InvalidAliasNameException = (function (_super) {
    __extends(InvalidAliasNameException, _super);
    function InvalidAliasNameException(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidAliasNameException", $fault: "client" }, opts)) || this;
        _this.name = "InvalidAliasNameException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidAliasNameException.prototype);
        return _this;
    }
    return InvalidAliasNameException;
}(__BaseException));
export { InvalidAliasNameException };
var LimitExceededException = (function (_super) {
    __extends(LimitExceededException, _super);
    function LimitExceededException(opts) {
        var _this = _super.call(this, __assign({ name: "LimitExceededException", $fault: "client" }, opts)) || this;
        _this.name = "LimitExceededException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, LimitExceededException.prototype);
        return _this;
    }
    return LimitExceededException;
}(__BaseException));
export { LimitExceededException };
export var CreateCustomKeyStoreRequest;
(function (CreateCustomKeyStoreRequest) {
    CreateCustomKeyStoreRequest.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.KeyStorePassword && { KeyStorePassword: SENSITIVE_STRING }))); };
})(CreateCustomKeyStoreRequest || (CreateCustomKeyStoreRequest = {}));
export var CreateCustomKeyStoreResponse;
(function (CreateCustomKeyStoreResponse) {
    CreateCustomKeyStoreResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateCustomKeyStoreResponse || (CreateCustomKeyStoreResponse = {}));
var CustomKeyStoreNameInUseException = (function (_super) {
    __extends(CustomKeyStoreNameInUseException, _super);
    function CustomKeyStoreNameInUseException(opts) {
        var _this = _super.call(this, __assign({ name: "CustomKeyStoreNameInUseException", $fault: "client" }, opts)) || this;
        _this.name = "CustomKeyStoreNameInUseException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, CustomKeyStoreNameInUseException.prototype);
        return _this;
    }
    return CustomKeyStoreNameInUseException;
}(__BaseException));
export { CustomKeyStoreNameInUseException };
var IncorrectTrustAnchorException = (function (_super) {
    __extends(IncorrectTrustAnchorException, _super);
    function IncorrectTrustAnchorException(opts) {
        var _this = _super.call(this, __assign({ name: "IncorrectTrustAnchorException", $fault: "client" }, opts)) || this;
        _this.name = "IncorrectTrustAnchorException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, IncorrectTrustAnchorException.prototype);
        return _this;
    }
    return IncorrectTrustAnchorException;
}(__BaseException));
export { IncorrectTrustAnchorException };
export var GrantConstraints;
(function (GrantConstraints) {
    GrantConstraints.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GrantConstraints || (GrantConstraints = {}));
export var GrantOperation;
(function (GrantOperation) {
    GrantOperation["CreateGrant"] = "CreateGrant";
    GrantOperation["Decrypt"] = "Decrypt";
    GrantOperation["DescribeKey"] = "DescribeKey";
    GrantOperation["Encrypt"] = "Encrypt";
    GrantOperation["GenerateDataKey"] = "GenerateDataKey";
    GrantOperation["GenerateDataKeyPair"] = "GenerateDataKeyPair";
    GrantOperation["GenerateDataKeyPairWithoutPlaintext"] = "GenerateDataKeyPairWithoutPlaintext";
    GrantOperation["GenerateDataKeyWithoutPlaintext"] = "GenerateDataKeyWithoutPlaintext";
    GrantOperation["GetPublicKey"] = "GetPublicKey";
    GrantOperation["ReEncryptFrom"] = "ReEncryptFrom";
    GrantOperation["ReEncryptTo"] = "ReEncryptTo";
    GrantOperation["RetireGrant"] = "RetireGrant";
    GrantOperation["Sign"] = "Sign";
    GrantOperation["Verify"] = "Verify";
})(GrantOperation || (GrantOperation = {}));
export var CreateGrantRequest;
(function (CreateGrantRequest) {
    CreateGrantRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateGrantRequest || (CreateGrantRequest = {}));
export var CreateGrantResponse;
(function (CreateGrantResponse) {
    CreateGrantResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateGrantResponse || (CreateGrantResponse = {}));
var DisabledException = (function (_super) {
    __extends(DisabledException, _super);
    function DisabledException(opts) {
        var _this = _super.call(this, __assign({ name: "DisabledException", $fault: "client" }, opts)) || this;
        _this.name = "DisabledException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, DisabledException.prototype);
        return _this;
    }
    return DisabledException;
}(__BaseException));
export { DisabledException };
var InvalidGrantTokenException = (function (_super) {
    __extends(InvalidGrantTokenException, _super);
    function InvalidGrantTokenException(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidGrantTokenException", $fault: "client" }, opts)) || this;
        _this.name = "InvalidGrantTokenException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidGrantTokenException.prototype);
        return _this;
    }
    return InvalidGrantTokenException;
}(__BaseException));
export { InvalidGrantTokenException };
export var CustomerMasterKeySpec;
(function (CustomerMasterKeySpec) {
    CustomerMasterKeySpec["ECC_NIST_P256"] = "ECC_NIST_P256";
    CustomerMasterKeySpec["ECC_NIST_P384"] = "ECC_NIST_P384";
    CustomerMasterKeySpec["ECC_NIST_P521"] = "ECC_NIST_P521";
    CustomerMasterKeySpec["ECC_SECG_P256K1"] = "ECC_SECG_P256K1";
    CustomerMasterKeySpec["RSA_2048"] = "RSA_2048";
    CustomerMasterKeySpec["RSA_3072"] = "RSA_3072";
    CustomerMasterKeySpec["RSA_4096"] = "RSA_4096";
    CustomerMasterKeySpec["SYMMETRIC_DEFAULT"] = "SYMMETRIC_DEFAULT";
})(CustomerMasterKeySpec || (CustomerMasterKeySpec = {}));
export var KeySpec;
(function (KeySpec) {
    KeySpec["ECC_NIST_P256"] = "ECC_NIST_P256";
    KeySpec["ECC_NIST_P384"] = "ECC_NIST_P384";
    KeySpec["ECC_NIST_P521"] = "ECC_NIST_P521";
    KeySpec["ECC_SECG_P256K1"] = "ECC_SECG_P256K1";
    KeySpec["RSA_2048"] = "RSA_2048";
    KeySpec["RSA_3072"] = "RSA_3072";
    KeySpec["RSA_4096"] = "RSA_4096";
    KeySpec["SYMMETRIC_DEFAULT"] = "SYMMETRIC_DEFAULT";
})(KeySpec || (KeySpec = {}));
export var KeyUsageType;
(function (KeyUsageType) {
    KeyUsageType["ENCRYPT_DECRYPT"] = "ENCRYPT_DECRYPT";
    KeyUsageType["SIGN_VERIFY"] = "SIGN_VERIFY";
})(KeyUsageType || (KeyUsageType = {}));
export var OriginType;
(function (OriginType) {
    OriginType["AWS_CLOUDHSM"] = "AWS_CLOUDHSM";
    OriginType["AWS_KMS"] = "AWS_KMS";
    OriginType["EXTERNAL"] = "EXTERNAL";
})(OriginType || (OriginType = {}));
export var Tag;
(function (Tag) {
    Tag.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Tag || (Tag = {}));
export var CreateKeyRequest;
(function (CreateKeyRequest) {
    CreateKeyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateKeyRequest || (CreateKeyRequest = {}));
export var EncryptionAlgorithmSpec;
(function (EncryptionAlgorithmSpec) {
    EncryptionAlgorithmSpec["RSAES_OAEP_SHA_1"] = "RSAES_OAEP_SHA_1";
    EncryptionAlgorithmSpec["RSAES_OAEP_SHA_256"] = "RSAES_OAEP_SHA_256";
    EncryptionAlgorithmSpec["SYMMETRIC_DEFAULT"] = "SYMMETRIC_DEFAULT";
})(EncryptionAlgorithmSpec || (EncryptionAlgorithmSpec = {}));
export var ExpirationModelType;
(function (ExpirationModelType) {
    ExpirationModelType["KEY_MATERIAL_DOES_NOT_EXPIRE"] = "KEY_MATERIAL_DOES_NOT_EXPIRE";
    ExpirationModelType["KEY_MATERIAL_EXPIRES"] = "KEY_MATERIAL_EXPIRES";
})(ExpirationModelType || (ExpirationModelType = {}));
export var KeyManagerType;
(function (KeyManagerType) {
    KeyManagerType["AWS"] = "AWS";
    KeyManagerType["CUSTOMER"] = "CUSTOMER";
})(KeyManagerType || (KeyManagerType = {}));
export var KeyState;
(function (KeyState) {
    KeyState["Creating"] = "Creating";
    KeyState["Disabled"] = "Disabled";
    KeyState["Enabled"] = "Enabled";
    KeyState["PendingDeletion"] = "PendingDeletion";
    KeyState["PendingImport"] = "PendingImport";
    KeyState["PendingReplicaDeletion"] = "PendingReplicaDeletion";
    KeyState["Unavailable"] = "Unavailable";
    KeyState["Updating"] = "Updating";
})(KeyState || (KeyState = {}));
export var MultiRegionKeyType;
(function (MultiRegionKeyType) {
    MultiRegionKeyType["PRIMARY"] = "PRIMARY";
    MultiRegionKeyType["REPLICA"] = "REPLICA";
})(MultiRegionKeyType || (MultiRegionKeyType = {}));
export var MultiRegionKey;
(function (MultiRegionKey) {
    MultiRegionKey.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(MultiRegionKey || (MultiRegionKey = {}));
export var MultiRegionConfiguration;
(function (MultiRegionConfiguration) {
    MultiRegionConfiguration.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(MultiRegionConfiguration || (MultiRegionConfiguration = {}));
export var SigningAlgorithmSpec;
(function (SigningAlgorithmSpec) {
    SigningAlgorithmSpec["ECDSA_SHA_256"] = "ECDSA_SHA_256";
    SigningAlgorithmSpec["ECDSA_SHA_384"] = "ECDSA_SHA_384";
    SigningAlgorithmSpec["ECDSA_SHA_512"] = "ECDSA_SHA_512";
    SigningAlgorithmSpec["RSASSA_PKCS1_V1_5_SHA_256"] = "RSASSA_PKCS1_V1_5_SHA_256";
    SigningAlgorithmSpec["RSASSA_PKCS1_V1_5_SHA_384"] = "RSASSA_PKCS1_V1_5_SHA_384";
    SigningAlgorithmSpec["RSASSA_PKCS1_V1_5_SHA_512"] = "RSASSA_PKCS1_V1_5_SHA_512";
    SigningAlgorithmSpec["RSASSA_PSS_SHA_256"] = "RSASSA_PSS_SHA_256";
    SigningAlgorithmSpec["RSASSA_PSS_SHA_384"] = "RSASSA_PSS_SHA_384";
    SigningAlgorithmSpec["RSASSA_PSS_SHA_512"] = "RSASSA_PSS_SHA_512";
})(SigningAlgorithmSpec || (SigningAlgorithmSpec = {}));
export var KeyMetadata;
(function (KeyMetadata) {
    KeyMetadata.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(KeyMetadata || (KeyMetadata = {}));
export var CreateKeyResponse;
(function (CreateKeyResponse) {
    CreateKeyResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateKeyResponse || (CreateKeyResponse = {}));
var MalformedPolicyDocumentException = (function (_super) {
    __extends(MalformedPolicyDocumentException, _super);
    function MalformedPolicyDocumentException(opts) {
        var _this = _super.call(this, __assign({ name: "MalformedPolicyDocumentException", $fault: "client" }, opts)) || this;
        _this.name = "MalformedPolicyDocumentException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, MalformedPolicyDocumentException.prototype);
        return _this;
    }
    return MalformedPolicyDocumentException;
}(__BaseException));
export { MalformedPolicyDocumentException };
var TagException = (function (_super) {
    __extends(TagException, _super);
    function TagException(opts) {
        var _this = _super.call(this, __assign({ name: "TagException", $fault: "client" }, opts)) || this;
        _this.name = "TagException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, TagException.prototype);
        return _this;
    }
    return TagException;
}(__BaseException));
export { TagException };
var UnsupportedOperationException = (function (_super) {
    __extends(UnsupportedOperationException, _super);
    function UnsupportedOperationException(opts) {
        var _this = _super.call(this, __assign({ name: "UnsupportedOperationException", $fault: "client" }, opts)) || this;
        _this.name = "UnsupportedOperationException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, UnsupportedOperationException.prototype);
        return _this;
    }
    return UnsupportedOperationException;
}(__BaseException));
export { UnsupportedOperationException };
var CustomKeyStoreHasCMKsException = (function (_super) {
    __extends(CustomKeyStoreHasCMKsException, _super);
    function CustomKeyStoreHasCMKsException(opts) {
        var _this = _super.call(this, __assign({ name: "CustomKeyStoreHasCMKsException", $fault: "client" }, opts)) || this;
        _this.name = "CustomKeyStoreHasCMKsException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, CustomKeyStoreHasCMKsException.prototype);
        return _this;
    }
    return CustomKeyStoreHasCMKsException;
}(__BaseException));
export { CustomKeyStoreHasCMKsException };
export var CustomKeyStoresListEntry;
(function (CustomKeyStoresListEntry) {
    CustomKeyStoresListEntry.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CustomKeyStoresListEntry || (CustomKeyStoresListEntry = {}));
export var DataKeyPairSpec;
(function (DataKeyPairSpec) {
    DataKeyPairSpec["ECC_NIST_P256"] = "ECC_NIST_P256";
    DataKeyPairSpec["ECC_NIST_P384"] = "ECC_NIST_P384";
    DataKeyPairSpec["ECC_NIST_P521"] = "ECC_NIST_P521";
    DataKeyPairSpec["ECC_SECG_P256K1"] = "ECC_SECG_P256K1";
    DataKeyPairSpec["RSA_2048"] = "RSA_2048";
    DataKeyPairSpec["RSA_3072"] = "RSA_3072";
    DataKeyPairSpec["RSA_4096"] = "RSA_4096";
})(DataKeyPairSpec || (DataKeyPairSpec = {}));
export var DataKeySpec;
(function (DataKeySpec) {
    DataKeySpec["AES_128"] = "AES_128";
    DataKeySpec["AES_256"] = "AES_256";
})(DataKeySpec || (DataKeySpec = {}));
export var DecryptRequest;
(function (DecryptRequest) {
    DecryptRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DecryptRequest || (DecryptRequest = {}));
export var DecryptResponse;
(function (DecryptResponse) {
    DecryptResponse.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Plaintext && { Plaintext: SENSITIVE_STRING }))); };
})(DecryptResponse || (DecryptResponse = {}));
var IncorrectKeyException = (function (_super) {
    __extends(IncorrectKeyException, _super);
    function IncorrectKeyException(opts) {
        var _this = _super.call(this, __assign({ name: "IncorrectKeyException", $fault: "client" }, opts)) || this;
        _this.name = "IncorrectKeyException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, IncorrectKeyException.prototype);
        return _this;
    }
    return IncorrectKeyException;
}(__BaseException));
export { IncorrectKeyException };
var InvalidCiphertextException = (function (_super) {
    __extends(InvalidCiphertextException, _super);
    function InvalidCiphertextException(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidCiphertextException", $fault: "client" }, opts)) || this;
        _this.name = "InvalidCiphertextException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidCiphertextException.prototype);
        return _this;
    }
    return InvalidCiphertextException;
}(__BaseException));
export { InvalidCiphertextException };
var InvalidKeyUsageException = (function (_super) {
    __extends(InvalidKeyUsageException, _super);
    function InvalidKeyUsageException(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidKeyUsageException", $fault: "client" }, opts)) || this;
        _this.name = "InvalidKeyUsageException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidKeyUsageException.prototype);
        return _this;
    }
    return InvalidKeyUsageException;
}(__BaseException));
export { InvalidKeyUsageException };
var KeyUnavailableException = (function (_super) {
    __extends(KeyUnavailableException, _super);
    function KeyUnavailableException(opts) {
        var _this = _super.call(this, __assign({ name: "KeyUnavailableException", $fault: "server" }, opts)) || this;
        _this.name = "KeyUnavailableException";
        _this.$fault = "server";
        Object.setPrototypeOf(_this, KeyUnavailableException.prototype);
        return _this;
    }
    return KeyUnavailableException;
}(__BaseException));
export { KeyUnavailableException };
export var DeleteAliasRequest;
(function (DeleteAliasRequest) {
    DeleteAliasRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteAliasRequest || (DeleteAliasRequest = {}));
export var DeleteCustomKeyStoreRequest;
(function (DeleteCustomKeyStoreRequest) {
    DeleteCustomKeyStoreRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteCustomKeyStoreRequest || (DeleteCustomKeyStoreRequest = {}));
export var DeleteCustomKeyStoreResponse;
(function (DeleteCustomKeyStoreResponse) {
    DeleteCustomKeyStoreResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteCustomKeyStoreResponse || (DeleteCustomKeyStoreResponse = {}));
export var DeleteImportedKeyMaterialRequest;
(function (DeleteImportedKeyMaterialRequest) {
    DeleteImportedKeyMaterialRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteImportedKeyMaterialRequest || (DeleteImportedKeyMaterialRequest = {}));
export var DescribeCustomKeyStoresRequest;
(function (DescribeCustomKeyStoresRequest) {
    DescribeCustomKeyStoresRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeCustomKeyStoresRequest || (DescribeCustomKeyStoresRequest = {}));
export var DescribeCustomKeyStoresResponse;
(function (DescribeCustomKeyStoresResponse) {
    DescribeCustomKeyStoresResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeCustomKeyStoresResponse || (DescribeCustomKeyStoresResponse = {}));
var InvalidMarkerException = (function (_super) {
    __extends(InvalidMarkerException, _super);
    function InvalidMarkerException(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidMarkerException", $fault: "client" }, opts)) || this;
        _this.name = "InvalidMarkerException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidMarkerException.prototype);
        return _this;
    }
    return InvalidMarkerException;
}(__BaseException));
export { InvalidMarkerException };
export var DescribeKeyRequest;
(function (DescribeKeyRequest) {
    DescribeKeyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeKeyRequest || (DescribeKeyRequest = {}));
export var DescribeKeyResponse;
(function (DescribeKeyResponse) {
    DescribeKeyResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeKeyResponse || (DescribeKeyResponse = {}));
export var DisableKeyRequest;
(function (DisableKeyRequest) {
    DisableKeyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DisableKeyRequest || (DisableKeyRequest = {}));
export var DisableKeyRotationRequest;
(function (DisableKeyRotationRequest) {
    DisableKeyRotationRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DisableKeyRotationRequest || (DisableKeyRotationRequest = {}));
export var DisconnectCustomKeyStoreRequest;
(function (DisconnectCustomKeyStoreRequest) {
    DisconnectCustomKeyStoreRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DisconnectCustomKeyStoreRequest || (DisconnectCustomKeyStoreRequest = {}));
export var DisconnectCustomKeyStoreResponse;
(function (DisconnectCustomKeyStoreResponse) {
    DisconnectCustomKeyStoreResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DisconnectCustomKeyStoreResponse || (DisconnectCustomKeyStoreResponse = {}));
export var EnableKeyRequest;
(function (EnableKeyRequest) {
    EnableKeyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(EnableKeyRequest || (EnableKeyRequest = {}));
export var EnableKeyRotationRequest;
(function (EnableKeyRotationRequest) {
    EnableKeyRotationRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(EnableKeyRotationRequest || (EnableKeyRotationRequest = {}));
export var EncryptRequest;
(function (EncryptRequest) {
    EncryptRequest.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Plaintext && { Plaintext: SENSITIVE_STRING }))); };
})(EncryptRequest || (EncryptRequest = {}));
export var EncryptResponse;
(function (EncryptResponse) {
    EncryptResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(EncryptResponse || (EncryptResponse = {}));
var ExpiredImportTokenException = (function (_super) {
    __extends(ExpiredImportTokenException, _super);
    function ExpiredImportTokenException(opts) {
        var _this = _super.call(this, __assign({ name: "ExpiredImportTokenException", $fault: "client" }, opts)) || this;
        _this.name = "ExpiredImportTokenException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, ExpiredImportTokenException.prototype);
        return _this;
    }
    return ExpiredImportTokenException;
}(__BaseException));
export { ExpiredImportTokenException };
export var GenerateDataKeyRequest;
(function (GenerateDataKeyRequest) {
    GenerateDataKeyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GenerateDataKeyRequest || (GenerateDataKeyRequest = {}));
export var GenerateDataKeyResponse;
(function (GenerateDataKeyResponse) {
    GenerateDataKeyResponse.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Plaintext && { Plaintext: SENSITIVE_STRING }))); };
})(GenerateDataKeyResponse || (GenerateDataKeyResponse = {}));
export var GenerateDataKeyPairRequest;
(function (GenerateDataKeyPairRequest) {
    GenerateDataKeyPairRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GenerateDataKeyPairRequest || (GenerateDataKeyPairRequest = {}));
export var GenerateDataKeyPairResponse;
(function (GenerateDataKeyPairResponse) {
    GenerateDataKeyPairResponse.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.PrivateKeyPlaintext && { PrivateKeyPlaintext: SENSITIVE_STRING }))); };
})(GenerateDataKeyPairResponse || (GenerateDataKeyPairResponse = {}));
export var GenerateDataKeyPairWithoutPlaintextRequest;
(function (GenerateDataKeyPairWithoutPlaintextRequest) {
    GenerateDataKeyPairWithoutPlaintextRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GenerateDataKeyPairWithoutPlaintextRequest || (GenerateDataKeyPairWithoutPlaintextRequest = {}));
export var GenerateDataKeyPairWithoutPlaintextResponse;
(function (GenerateDataKeyPairWithoutPlaintextResponse) {
    GenerateDataKeyPairWithoutPlaintextResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GenerateDataKeyPairWithoutPlaintextResponse || (GenerateDataKeyPairWithoutPlaintextResponse = {}));
export var GenerateDataKeyWithoutPlaintextRequest;
(function (GenerateDataKeyWithoutPlaintextRequest) {
    GenerateDataKeyWithoutPlaintextRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GenerateDataKeyWithoutPlaintextRequest || (GenerateDataKeyWithoutPlaintextRequest = {}));
export var GenerateDataKeyWithoutPlaintextResponse;
(function (GenerateDataKeyWithoutPlaintextResponse) {
    GenerateDataKeyWithoutPlaintextResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GenerateDataKeyWithoutPlaintextResponse || (GenerateDataKeyWithoutPlaintextResponse = {}));
export var GenerateRandomRequest;
(function (GenerateRandomRequest) {
    GenerateRandomRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GenerateRandomRequest || (GenerateRandomRequest = {}));
export var GenerateRandomResponse;
(function (GenerateRandomResponse) {
    GenerateRandomResponse.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Plaintext && { Plaintext: SENSITIVE_STRING }))); };
})(GenerateRandomResponse || (GenerateRandomResponse = {}));
export var GetKeyPolicyRequest;
(function (GetKeyPolicyRequest) {
    GetKeyPolicyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetKeyPolicyRequest || (GetKeyPolicyRequest = {}));
export var GetKeyPolicyResponse;
(function (GetKeyPolicyResponse) {
    GetKeyPolicyResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetKeyPolicyResponse || (GetKeyPolicyResponse = {}));
export var GetKeyRotationStatusRequest;
(function (GetKeyRotationStatusRequest) {
    GetKeyRotationStatusRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetKeyRotationStatusRequest || (GetKeyRotationStatusRequest = {}));
export var GetKeyRotationStatusResponse;
(function (GetKeyRotationStatusResponse) {
    GetKeyRotationStatusResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetKeyRotationStatusResponse || (GetKeyRotationStatusResponse = {}));
export var WrappingKeySpec;
(function (WrappingKeySpec) {
    WrappingKeySpec["RSA_2048"] = "RSA_2048";
})(WrappingKeySpec || (WrappingKeySpec = {}));
export var GetParametersForImportRequest;
(function (GetParametersForImportRequest) {
    GetParametersForImportRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetParametersForImportRequest || (GetParametersForImportRequest = {}));
export var GetParametersForImportResponse;
(function (GetParametersForImportResponse) {
    GetParametersForImportResponse.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.PublicKey && { PublicKey: SENSITIVE_STRING }))); };
})(GetParametersForImportResponse || (GetParametersForImportResponse = {}));
export var GetPublicKeyRequest;
(function (GetPublicKeyRequest) {
    GetPublicKeyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetPublicKeyRequest || (GetPublicKeyRequest = {}));
export var GetPublicKeyResponse;
(function (GetPublicKeyResponse) {
    GetPublicKeyResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetPublicKeyResponse || (GetPublicKeyResponse = {}));
export var GrantListEntry;
(function (GrantListEntry) {
    GrantListEntry.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GrantListEntry || (GrantListEntry = {}));
export var ImportKeyMaterialRequest;
(function (ImportKeyMaterialRequest) {
    ImportKeyMaterialRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ImportKeyMaterialRequest || (ImportKeyMaterialRequest = {}));
export var ImportKeyMaterialResponse;
(function (ImportKeyMaterialResponse) {
    ImportKeyMaterialResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ImportKeyMaterialResponse || (ImportKeyMaterialResponse = {}));
var IncorrectKeyMaterialException = (function (_super) {
    __extends(IncorrectKeyMaterialException, _super);
    function IncorrectKeyMaterialException(opts) {
        var _this = _super.call(this, __assign({ name: "IncorrectKeyMaterialException", $fault: "client" }, opts)) || this;
        _this.name = "IncorrectKeyMaterialException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, IncorrectKeyMaterialException.prototype);
        return _this;
    }
    return IncorrectKeyMaterialException;
}(__BaseException));
export { IncorrectKeyMaterialException };
var InvalidImportTokenException = (function (_super) {
    __extends(InvalidImportTokenException, _super);
    function InvalidImportTokenException(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidImportTokenException", $fault: "client" }, opts)) || this;
        _this.name = "InvalidImportTokenException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidImportTokenException.prototype);
        return _this;
    }
    return InvalidImportTokenException;
}(__BaseException));
export { InvalidImportTokenException };
var InvalidGrantIdException = (function (_super) {
    __extends(InvalidGrantIdException, _super);
    function InvalidGrantIdException(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidGrantIdException", $fault: "client" }, opts)) || this;
        _this.name = "InvalidGrantIdException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidGrantIdException.prototype);
        return _this;
    }
    return InvalidGrantIdException;
}(__BaseException));
export { InvalidGrantIdException };
export var KeyListEntry;
(function (KeyListEntry) {
    KeyListEntry.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(KeyListEntry || (KeyListEntry = {}));
var KMSInvalidSignatureException = (function (_super) {
    __extends(KMSInvalidSignatureException, _super);
    function KMSInvalidSignatureException(opts) {
        var _this = _super.call(this, __assign({ name: "KMSInvalidSignatureException", $fault: "client" }, opts)) || this;
        _this.name = "KMSInvalidSignatureException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, KMSInvalidSignatureException.prototype);
        return _this;
    }
    return KMSInvalidSignatureException;
}(__BaseException));
export { KMSInvalidSignatureException };
export var ListAliasesRequest;
(function (ListAliasesRequest) {
    ListAliasesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListAliasesRequest || (ListAliasesRequest = {}));
export var ListAliasesResponse;
(function (ListAliasesResponse) {
    ListAliasesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListAliasesResponse || (ListAliasesResponse = {}));
export var ListGrantsRequest;
(function (ListGrantsRequest) {
    ListGrantsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListGrantsRequest || (ListGrantsRequest = {}));
export var ListGrantsResponse;
(function (ListGrantsResponse) {
    ListGrantsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListGrantsResponse || (ListGrantsResponse = {}));
export var ListKeyPoliciesRequest;
(function (ListKeyPoliciesRequest) {
    ListKeyPoliciesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListKeyPoliciesRequest || (ListKeyPoliciesRequest = {}));
export var ListKeyPoliciesResponse;
(function (ListKeyPoliciesResponse) {
    ListKeyPoliciesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListKeyPoliciesResponse || (ListKeyPoliciesResponse = {}));
export var ListKeysRequest;
(function (ListKeysRequest) {
    ListKeysRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListKeysRequest || (ListKeysRequest = {}));
export var ListKeysResponse;
(function (ListKeysResponse) {
    ListKeysResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListKeysResponse || (ListKeysResponse = {}));
export var ListResourceTagsRequest;
(function (ListResourceTagsRequest) {
    ListResourceTagsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListResourceTagsRequest || (ListResourceTagsRequest = {}));
export var ListResourceTagsResponse;
(function (ListResourceTagsResponse) {
    ListResourceTagsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListResourceTagsResponse || (ListResourceTagsResponse = {}));
export var ListRetirableGrantsRequest;
(function (ListRetirableGrantsRequest) {
    ListRetirableGrantsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListRetirableGrantsRequest || (ListRetirableGrantsRequest = {}));
export var MessageType;
(function (MessageType) {
    MessageType["DIGEST"] = "DIGEST";
    MessageType["RAW"] = "RAW";
})(MessageType || (MessageType = {}));
export var PutKeyPolicyRequest;
(function (PutKeyPolicyRequest) {
    PutKeyPolicyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PutKeyPolicyRequest || (PutKeyPolicyRequest = {}));
export var ReEncryptRequest;
(function (ReEncryptRequest) {
    ReEncryptRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReEncryptRequest || (ReEncryptRequest = {}));
export var ReEncryptResponse;
(function (ReEncryptResponse) {
    ReEncryptResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReEncryptResponse || (ReEncryptResponse = {}));
export var ReplicateKeyRequest;
(function (ReplicateKeyRequest) {
    ReplicateKeyRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReplicateKeyRequest || (ReplicateKeyRequest = {}));
export var ReplicateKeyResponse;
(function (ReplicateKeyResponse) {
    ReplicateKeyResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReplicateKeyResponse || (ReplicateKeyResponse = {}));
export var RetireGrantRequest;
(function (RetireGrantRequest) {
    RetireGrantRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RetireGrantRequest || (RetireGrantRequest = {}));
export var RevokeGrantRequest;
(function (RevokeGrantRequest) {
    RevokeGrantRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RevokeGrantRequest || (RevokeGrantRequest = {}));
export var ScheduleKeyDeletionRequest;
(function (ScheduleKeyDeletionRequest) {
    ScheduleKeyDeletionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ScheduleKeyDeletionRequest || (ScheduleKeyDeletionRequest = {}));
export var ScheduleKeyDeletionResponse;
(function (ScheduleKeyDeletionResponse) {
    ScheduleKeyDeletionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ScheduleKeyDeletionResponse || (ScheduleKeyDeletionResponse = {}));
export var SignRequest;
(function (SignRequest) {
    SignRequest.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Message && { Message: SENSITIVE_STRING }))); };
})(SignRequest || (SignRequest = {}));
export var SignResponse;
(function (SignResponse) {
    SignResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SignResponse || (SignResponse = {}));
export var TagResourceRequest;
(function (TagResourceRequest) {
    TagResourceRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TagResourceRequest || (TagResourceRequest = {}));
export var UntagResourceRequest;
(function (UntagResourceRequest) {
    UntagResourceRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UntagResourceRequest || (UntagResourceRequest = {}));
export var UpdateAliasRequest;
(function (UpdateAliasRequest) {
    UpdateAliasRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateAliasRequest || (UpdateAliasRequest = {}));
export var UpdateCustomKeyStoreRequest;
(function (UpdateCustomKeyStoreRequest) {
    UpdateCustomKeyStoreRequest.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.KeyStorePassword && { KeyStorePassword: SENSITIVE_STRING }))); };
})(UpdateCustomKeyStoreRequest || (UpdateCustomKeyStoreRequest = {}));
export var UpdateCustomKeyStoreResponse;
(function (UpdateCustomKeyStoreResponse) {
    UpdateCustomKeyStoreResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateCustomKeyStoreResponse || (UpdateCustomKeyStoreResponse = {}));
export var UpdateKeyDescriptionRequest;
(function (UpdateKeyDescriptionRequest) {
    UpdateKeyDescriptionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateKeyDescriptionRequest || (UpdateKeyDescriptionRequest = {}));
export var UpdatePrimaryRegionRequest;
(function (UpdatePrimaryRegionRequest) {
    UpdatePrimaryRegionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdatePrimaryRegionRequest || (UpdatePrimaryRegionRequest = {}));
export var VerifyRequest;
(function (VerifyRequest) {
    VerifyRequest.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Message && { Message: SENSITIVE_STRING }))); };
})(VerifyRequest || (VerifyRequest = {}));
export var VerifyResponse;
(function (VerifyResponse) {
    VerifyResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(VerifyResponse || (VerifyResponse = {}));
